// @vuepic/vue-datepicker/dist/vue-datepicker.js
import { openBlock as $, createElementBlock as O, createElementVNode as ue, unref as m, reactive as Nt, computed as H, ref as z, toRef as Kt, watch as mt, h as Ya, render as Wn, defineComponent as Be, renderSlot as ee, createCommentVNode as I, normalizeClass as he, withKeys as oe, createBlock as ve, withModifiers as pe, onMounted as Ue, normalizeStyle as Ge, normalizeProps as Re, mergeProps as Ee, Fragment as ce, createTextVNode as nt, toDisplayString as Ie, onBeforeUpdate as Ba, nextTick as at, onUnmounted as _n, renderList as Me, withDirectives as Ut, vShow as Wt, withCtx as de, createVNode as Ke, Transition as wt, createSlots as Fe, useSlots as Ot, guardReactiveProps as He, resolveDynamicComponent as Sn, getCurrentScope as Ea, onScopeDispose as Fa, isRef as zn, Teleport as Va } from "vue";
import { format as it, isBefore as _t, isEqual as ht, isAfter as St, set as De, setHours as ia, setMinutes as da, setSeconds as ca, setMilliseconds as Pn, getYear as fe, getMonth as ge, setMonth as Ha, setYear as dt, addMonths as ct, subMonths as Pt, isValid as zt, parseISO as La, eachDayOfInterval as fa, getHours as qe, getMinutes as xe, getSeconds as pt, startOfWeek as va, endOfWeek as Ua, parse as bn, isDate as Wa, addYears as ma, subYears as ga, addDays as ut, getDay as za, differenceInCalendarDays as ja, add as ya, sub as Ka, getWeek as Ga, getISOWeek as Za, differenceInMonths as qa } from "date-fns";
function It() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M29.333 8c0-2.208-1.792-4-4-4h-18.667c-2.208 0-4 1.792-4 4v18.667c0 2.208 1.792 4 4 4h18.667c2.208 0 4-1.792 4-4v-18.667zM26.667 8v18.667c0 0.736-0.597 1.333-1.333 1.333 0 0-18.667 0-18.667 0-0.736 0-1.333-0.597-1.333-1.333 0 0 0-18.667 0-18.667 0-0.736 0.597-1.333 1.333-1.333 0 0 18.667 0 18.667 0 0.736 0 1.333 0.597 1.333 1.333z"
      }),
      ue("path", {
        d: "M20 2.667v5.333c0 0.736 0.597 1.333 1.333 1.333s1.333-0.597 1.333-1.333v-5.333c0-0.736-0.597-1.333-1.333-1.333s-1.333 0.597-1.333 1.333z"
      }),
      ue("path", {
        d: "M9.333 2.667v5.333c0 0.736 0.597 1.333 1.333 1.333s1.333-0.597 1.333-1.333v-5.333c0-0.736-0.597-1.333-1.333-1.333s-1.333 0.597-1.333 1.333z"
      }),
      ue("path", {
        d: "M4 14.667h24c0.736 0 1.333-0.597 1.333-1.333s-0.597-1.333-1.333-1.333h-24c-0.736 0-1.333 0.597-1.333 1.333s0.597 1.333 1.333 1.333z"
      })
    ]
  );
}
It.compatConfig = {
  MODE: 3
};
function ha() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M23.057 7.057l-16 16c-0.52 0.52-0.52 1.365 0 1.885s1.365 0.52 1.885 0l16-16c0.52-0.52 0.52-1.365 0-1.885s-1.365-0.52-1.885 0z"
      }),
      ue("path", {
        d: "M7.057 8.943l16 16c0.52 0.52 1.365 0.52 1.885 0s0.52-1.365 0-1.885l-16-16c-0.52-0.52-1.365-0.52-1.885 0s-0.52 1.365 0 1.885z"
      })
    ]
  );
}
ha.compatConfig = {
  MODE: 3
};
function Cn() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M20.943 23.057l-7.057-7.057c0 0 7.057-7.057 7.057-7.057 0.52-0.52 0.52-1.365 0-1.885s-1.365-0.52-1.885 0l-8 8c-0.521 0.521-0.521 1.365 0 1.885l8 8c0.52 0.52 1.365 0.52 1.885 0s0.52-1.365 0-1.885z"
      })
    ]
  );
}
Cn.compatConfig = {
  MODE: 3
};
function Rn() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M12.943 24.943l8-8c0.521-0.521 0.521-1.365 0-1.885l-8-8c-0.52-0.52-1.365-0.52-1.885 0s-0.52 1.365 0 1.885l7.057 7.057c0 0-7.057 7.057-7.057 7.057-0.52 0.52-0.52 1.365 0 1.885s1.365 0.52 1.885 0z"
      })
    ]
  );
}
Rn.compatConfig = {
  MODE: 3
};
function Nn() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M16 1.333c-8.095 0-14.667 6.572-14.667 14.667s6.572 14.667 14.667 14.667c8.095 0 14.667-6.572 14.667-14.667s-6.572-14.667-14.667-14.667zM16 4c6.623 0 12 5.377 12 12s-5.377 12-12 12c-6.623 0-12-5.377-12-12s5.377-12 12-12z"
      }),
      ue("path", {
        d: "M14.667 8v8c0 0.505 0.285 0.967 0.737 1.193l5.333 2.667c0.658 0.329 1.46 0.062 1.789-0.596s0.062-1.46-0.596-1.789l-4.596-2.298c0 0 0-7.176 0-7.176 0-0.736-0.597-1.333-1.333-1.333s-1.333 0.597-1.333 1.333z"
      })
    ]
  );
}
Nn.compatConfig = {
  MODE: 3
};
function On() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M24.943 19.057l-8-8c-0.521-0.521-1.365-0.521-1.885 0l-8 8c-0.52 0.52-0.52 1.365 0 1.885s1.365 0.52 1.885 0l7.057-7.057c0 0 7.057 7.057 7.057 7.057 0.52 0.52 1.365 0.52 1.885 0s0.52-1.365 0-1.885z"
      })
    ]
  );
}
On.compatConfig = {
  MODE: 3
};
function In() {
  return $(), O(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      viewBox: "0 0 32 32",
      fill: "currentColor",
      "aria-hidden": "true",
      class: "dp__icon"
    },
    [
      ue("path", {
        d: "M7.057 12.943l8 8c0.521 0.521 1.365 0.521 1.885 0l8-8c0.52-0.52 0.52-1.365 0-1.885s-1.365-0.52-1.885 0l-7.057 7.057c0 0-7.057-7.057-7.057-7.057-0.52-0.52-1.365-0.52-1.885 0s-0.52 1.365 0 1.885z"
      })
    ]
  );
}
In.compatConfig = {
  MODE: 3
};
function Yn(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var pa = { exports: {} };
(function(e) {
  function n(a) {
    return a && a.__esModule ? a : {
      default: a
    };
  }
  e.exports = n, e.exports.__esModule = !0, e.exports.default = e.exports;
})(pa);
var xa = pa.exports, kn = { exports: {} };
(function(e, n) {
  Object.defineProperty(n, "__esModule", {
    value: !0
  }), n.default = a;
  function a(t) {
    if (t === null || t === !0 || t === !1)
      return NaN;
    var o = Number(t);
    return isNaN(o) ? o : o < 0 ? Math.ceil(o) : Math.floor(o);
  }
  e.exports = n.default;
})(kn, kn.exports);
var Ja = kn.exports;
const Xa = /* @__PURE__ */ Yn(Ja);
var wn = { exports: {} };
(function(e, n) {
  Object.defineProperty(n, "__esModule", {
    value: !0
  }), n.default = a;
  function a(t) {
    var o = new Date(Date.UTC(t.getFullYear(), t.getMonth(), t.getDate(), t.getHours(), t.getMinutes(), t.getSeconds(), t.getMilliseconds()));
    return o.setUTCFullYear(t.getFullYear()), t.getTime() - o.getTime();
  }
  e.exports = n.default;
})(wn, wn.exports);
var Qa = wn.exports;
const jn = /* @__PURE__ */ Yn(Qa);
function er(e, n) {
  var a = rr(n);
  return a.formatToParts ? nr(a, e) : ar(a, e);
}
var tr = {
  year: 0,
  month: 1,
  day: 2,
  hour: 3,
  minute: 4,
  second: 5
};
function nr(e, n) {
  try {
    for (var a = e.formatToParts(n), t = [], o = 0; o < a.length; o++) {
      var l = tr[a[o].type];
      l >= 0 && (t[l] = parseInt(a[o].value, 10));
    }
    return t;
  } catch (c) {
    if (c instanceof RangeError)
      return [NaN];
    throw c;
  }
}
function ar(e, n) {
  var a = e.format(n).replace(/\u200E/g, ""), t = /(\d+)\/(\d+)\/(\d+),? (\d+):(\d+):(\d+)/.exec(a);
  return [t[3], t[1], t[2], t[4], t[5], t[6]];
}
var on = {};
function rr(e) {
  if (!on[e]) {
    var n = new Intl.DateTimeFormat("en-US", {
      hour12: !1,
      timeZone: "America/New_York",
      year: "numeric",
      month: "numeric",
      day: "2-digit",
      hour: "2-digit",
      minute: "2-digit",
      second: "2-digit"
    }).format(/* @__PURE__ */ new Date("2014-06-25T04:00:00.123Z")), a = n === "06/25/2014, 00:00:00" || n === "‎06‎/‎25‎/‎2014‎ ‎00‎:‎00‎:‎00";
    on[e] = a ? new Intl.DateTimeFormat("en-US", {
      hour12: !1,
      timeZone: e,
      year: "numeric",
      month: "numeric",
      day: "2-digit",
      hour: "2-digit",
      minute: "2-digit",
      second: "2-digit"
    }) : new Intl.DateTimeFormat("en-US", {
      hourCycle: "h23",
      timeZone: e,
      year: "numeric",
      month: "numeric",
      day: "2-digit",
      hour: "2-digit",
      minute: "2-digit",
      second: "2-digit"
    });
  }
  return on[e];
}
function Bn(e, n, a, t, o, l, c) {
  var b = /* @__PURE__ */ new Date(0);
  return b.setUTCFullYear(e, n, a), b.setUTCHours(t, o, l, c), b;
}
var Kn = 36e5, lr = 6e4, sn = {
  timezone: /([Z+-].*)$/,
  timezoneZ: /^(Z)$/,
  timezoneHH: /^([+-]\d{2})$/,
  timezoneHHMM: /^([+-]\d{2}):?(\d{2})$/
};
function En(e, n, a) {
  var t, o;
  if (!e || (t = sn.timezoneZ.exec(e), t))
    return 0;
  var l;
  if (t = sn.timezoneHH.exec(e), t)
    return l = parseInt(t[1], 10), Gn(l) ? -(l * Kn) : NaN;
  if (t = sn.timezoneHHMM.exec(e), t) {
    l = parseInt(t[1], 10);
    var c = parseInt(t[2], 10);
    return Gn(l, c) ? (o = Math.abs(l) * Kn + c * lr, l > 0 ? -o : o) : NaN;
  }
  if (ur(e)) {
    n = new Date(n || Date.now());
    var b = a ? n : or(n), p = Dn(b, e), M = a ? p : sr(n, p, e);
    return -M;
  }
  return NaN;
}
function or(e) {
  return Bn(
    e.getFullYear(),
    e.getMonth(),
    e.getDate(),
    e.getHours(),
    e.getMinutes(),
    e.getSeconds(),
    e.getMilliseconds()
  );
}
function Dn(e, n) {
  var a = er(e, n), t = Bn(
    a[0],
    a[1] - 1,
    a[2],
    a[3] % 24,
    a[4],
    a[5],
    0
  ).getTime(), o = e.getTime(), l = o % 1e3;
  return o -= l >= 0 ? l : 1e3 + l, t - o;
}
function sr(e, n, a) {
  var t = e.getTime(), o = t - n, l = Dn(new Date(o), a);
  if (n === l)
    return n;
  o -= l - n;
  var c = Dn(new Date(o), a);
  return l === c ? l : Math.max(l, c);
}
function Gn(e, n) {
  return -23 <= e && e <= 23 && (n == null || 0 <= n && n <= 59);
}
var Zn = {};
function ur(e) {
  if (Zn[e])
    return !0;
  try {
    return new Intl.DateTimeFormat(void 0, { timeZone: e }), Zn[e] = !0, !0;
  } catch {
    return !1;
  }
}
var ba = /(Z|[+-]\d{2}(?::?\d{2})?| UTC| [a-zA-Z]+\/[a-zA-Z_]+(?:\/[a-zA-Z_]+)?)$/, un = 36e5, qn = 6e4, ir = 2, Ye = {
  dateTimePattern: /^([0-9W+-]+)(T| )(.*)/,
  datePattern: /^([0-9W+-]+)(.*)/,
  plainTime: /:/,
  // year tokens
  YY: /^(\d{2})$/,
  YYY: [
    /^([+-]\d{2})$/,
    // 0 additional digits
    /^([+-]\d{3})$/,
    // 1 additional digit
    /^([+-]\d{4})$/
    // 2 additional digits
  ],
  YYYY: /^(\d{4})/,
  YYYYY: [
    /^([+-]\d{4})/,
    // 0 additional digits
    /^([+-]\d{5})/,
    // 1 additional digit
    /^([+-]\d{6})/
    // 2 additional digits
  ],
  // date tokens
  MM: /^-(\d{2})$/,
  DDD: /^-?(\d{3})$/,
  MMDD: /^-?(\d{2})-?(\d{2})$/,
  Www: /^-?W(\d{2})$/,
  WwwD: /^-?W(\d{2})-?(\d{1})$/,
  HH: /^(\d{2}([.,]\d*)?)$/,
  HHMM: /^(\d{2}):?(\d{2}([.,]\d*)?)$/,
  HHMMSS: /^(\d{2}):?(\d{2}):?(\d{2}([.,]\d*)?)$/,
  // time zone tokens (to identify the presence of a tz)
  timeZone: ba
};
function Mn(e, n) {
  if (arguments.length < 1)
    throw new TypeError("1 argument required, but only " + arguments.length + " present");
  if (e === null)
    return /* @__PURE__ */ new Date(NaN);
  var a = n || {}, t = a.additionalDigits == null ? ir : Xa(a.additionalDigits);
  if (t !== 2 && t !== 1 && t !== 0)
    throw new RangeError("additionalDigits must be 0, 1 or 2");
  if (e instanceof Date || typeof e == "object" && Object.prototype.toString.call(e) === "[object Date]")
    return new Date(e.getTime());
  if (typeof e == "number" || Object.prototype.toString.call(e) === "[object Number]")
    return new Date(e);
  if (!(typeof e == "string" || Object.prototype.toString.call(e) === "[object String]"))
    return /* @__PURE__ */ new Date(NaN);
  var o = dr(e), l = cr(o.date, t), c = l.year, b = l.restDateString, p = fr(b, c);
  if (isNaN(p))
    return /* @__PURE__ */ new Date(NaN);
  if (p) {
    var M = p.getTime(), k = 0, T;
    if (o.time && (k = vr(o.time), isNaN(k)))
      return /* @__PURE__ */ new Date(NaN);
    if (o.timeZone || a.timeZone) {
      if (T = En(o.timeZone || a.timeZone, new Date(M + k)), isNaN(T))
        return /* @__PURE__ */ new Date(NaN);
    } else
      T = jn(new Date(M + k)), T = jn(new Date(M + k + T));
    return new Date(M + k + T);
  } else
    return /* @__PURE__ */ new Date(NaN);
}
function dr(e) {
  var n = {}, a = Ye.dateTimePattern.exec(e), t;
  if (a ? (n.date = a[1], t = a[3]) : (a = Ye.datePattern.exec(e), a ? (n.date = a[1], t = a[2]) : (n.date = null, t = e)), t) {
    var o = Ye.timeZone.exec(t);
    o ? (n.time = t.replace(o[1], ""), n.timeZone = o[1].trim()) : n.time = t;
  }
  return n;
}
function cr(e, n) {
  var a = Ye.YYY[n], t = Ye.YYYYY[n], o;
  if (o = Ye.YYYY.exec(e) || t.exec(e), o) {
    var l = o[1];
    return {
      year: parseInt(l, 10),
      restDateString: e.slice(l.length)
    };
  }
  if (o = Ye.YY.exec(e) || a.exec(e), o) {
    var c = o[1];
    return {
      year: parseInt(c, 10) * 100,
      restDateString: e.slice(c.length)
    };
  }
  return {
    year: null
  };
}
function fr(e, n) {
  if (n === null)
    return null;
  var a, t, o, l;
  if (e.length === 0)
    return t = /* @__PURE__ */ new Date(0), t.setUTCFullYear(n), t;
  if (a = Ye.MM.exec(e), a)
    return t = /* @__PURE__ */ new Date(0), o = parseInt(a[1], 10) - 1, Jn(n, o) ? (t.setUTCFullYear(n, o), t) : /* @__PURE__ */ new Date(NaN);
  if (a = Ye.DDD.exec(e), a) {
    t = /* @__PURE__ */ new Date(0);
    var c = parseInt(a[1], 10);
    return yr(n, c) ? (t.setUTCFullYear(n, 0, c), t) : /* @__PURE__ */ new Date(NaN);
  }
  if (a = Ye.MMDD.exec(e), a) {
    t = /* @__PURE__ */ new Date(0), o = parseInt(a[1], 10) - 1;
    var b = parseInt(a[2], 10);
    return Jn(n, o, b) ? (t.setUTCFullYear(n, o, b), t) : /* @__PURE__ */ new Date(NaN);
  }
  if (a = Ye.Www.exec(e), a)
    return l = parseInt(a[1], 10) - 1, Xn(n, l) ? xn(n, l) : /* @__PURE__ */ new Date(NaN);
  if (a = Ye.WwwD.exec(e), a) {
    l = parseInt(a[1], 10) - 1;
    var p = parseInt(a[2], 10) - 1;
    return Xn(n, l, p) ? xn(n, l, p) : /* @__PURE__ */ new Date(NaN);
  }
  return null;
}
function vr(e) {
  var n, a, t;
  if (n = Ye.HH.exec(e), n)
    return a = parseFloat(n[1].replace(",", ".")), dn(a) ? a % 24 * un : NaN;
  if (n = Ye.HHMM.exec(e), n)
    return a = parseInt(n[1], 10), t = parseFloat(n[2].replace(",", ".")), dn(a, t) ? a % 24 * un + t * qn : NaN;
  if (n = Ye.HHMMSS.exec(e), n) {
    a = parseInt(n[1], 10), t = parseInt(n[2], 10);
    var o = parseFloat(n[3].replace(",", "."));
    return dn(a, t, o) ? a % 24 * un + t * qn + o * 1e3 : NaN;
  }
  return null;
}
function xn(e, n, a) {
  n = n || 0, a = a || 0;
  var t = /* @__PURE__ */ new Date(0);
  t.setUTCFullYear(e, 0, 4);
  var o = t.getUTCDay() || 7, l = n * 7 + a + 1 - o;
  return t.setUTCDate(t.getUTCDate() + l), t;
}
var mr = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], gr = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
function ka(e) {
  return e % 400 === 0 || e % 4 === 0 && e % 100 !== 0;
}
function Jn(e, n, a) {
  if (n < 0 || n > 11)
    return !1;
  if (a != null) {
    if (a < 1)
      return !1;
    var t = ka(e);
    if (t && a > gr[n] || !t && a > mr[n])
      return !1;
  }
  return !0;
}
function yr(e, n) {
  if (n < 1)
    return !1;
  var a = ka(e);
  return !(a && n > 366 || !a && n > 365);
}
function Xn(e, n, a) {
  return !(n < 0 || n > 52 || a != null && (a < 0 || a > 6));
}
function dn(e, n, a) {
  return !(e != null && (e < 0 || e >= 25) || n != null && (n < 0 || n >= 60) || a != null && (a < 0 || a >= 60));
}
var $n = { exports: {} }, Tn = { exports: {} };
(function(e, n) {
  Object.defineProperty(n, "__esModule", {
    value: !0
  }), n.default = a;
  function a(t, o) {
    if (t == null)
      throw new TypeError("assign requires that input parameter not be null or undefined");
    for (var l in o)
      Object.prototype.hasOwnProperty.call(o, l) && (t[l] = o[l]);
    return t;
  }
  e.exports = n.default;
})(Tn, Tn.exports);
var hr = Tn.exports;
(function(e, n) {
  var a = xa.default;
  Object.defineProperty(n, "__esModule", {
    value: !0
  }), n.default = o;
  var t = a(hr);
  function o(l) {
    return (0, t.default)({}, l);
  }
  e.exports = n.default;
})($n, $n.exports);
var pr = $n.exports;
const br = /* @__PURE__ */ Yn(pr);
function kr(e, n, a) {
  var t = Mn(e, a), o = En(n, t, !0), l = new Date(t.getTime() - o), c = /* @__PURE__ */ new Date(0);
  return c.setFullYear(l.getUTCFullYear(), l.getUTCMonth(), l.getUTCDate()), c.setHours(l.getUTCHours(), l.getUTCMinutes(), l.getUTCSeconds(), l.getUTCMilliseconds()), c;
}
function wr(e, n, a) {
  if (typeof e == "string" && !e.match(ba)) {
    var t = br(a);
    return t.timeZone = n, Mn(e, t);
  }
  var o = Mn(e, a), l = Bn(
    o.getFullYear(),
    o.getMonth(),
    o.getDate(),
    o.getHours(),
    o.getMinutes(),
    o.getSeconds(),
    o.getMilliseconds()
  ).getTime(), c = En(n, new Date(l));
  return new Date(l + c);
}
function Qn(e) {
  return (n) => new Intl.DateTimeFormat(e, { weekday: "short", timeZone: "UTC" }).format(/* @__PURE__ */ new Date(`2017-01-0${n}T00:00:00+00:00`)).slice(0, 2);
}
function Dr(e) {
  return (n) => it(/* @__PURE__ */ new Date(`2017-01-0${n}T00:00:00+00:00`), "EEEEEE", { locale: e });
}
const Mr = (e, n, a) => {
  const t = [1, 2, 3, 4, 5, 6, 7];
  let o;
  if (e !== null)
    try {
      o = t.map(Dr(e));
    } catch {
      o = t.map(Qn(n));
    }
  else
    o = t.map(Qn(n));
  const l = o.slice(0, a), c = o.slice(a + 1, o.length);
  return [o[a]].concat(...c).concat(...l);
}, Fn = (e, n) => {
  const a = [];
  for (let t = +e[0]; t <= +e[1]; t++)
    a.push({ value: +t, text: `${t}` });
  return n ? a.reverse() : a;
}, wa = (e, n, a) => {
  const t = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12].map((l) => {
    const c = l < 10 ? `0${l}` : l;
    return /* @__PURE__ */ new Date(`2017-${c}-01T00:00:00+00:00`);
  });
  if (e !== null)
    try {
      const l = a === "long" ? "MMMM" : "MMM";
      return t.map((c, b) => {
        const p = it(c, l, { locale: e });
        return {
          text: p.charAt(0).toUpperCase() + p.substring(1),
          value: b
        };
      });
    } catch {
    }
  const o = new Intl.DateTimeFormat(n, { month: a, timeZone: "UTC" });
  return t.map((l, c) => {
    const b = o.format(l);
    return {
      text: b.charAt(0).toUpperCase() + b.substring(1),
      value: c
    };
  });
}, $r = (e) => [12, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11][e], Ae = (e) => {
  const n = m(e);
  return n != null && n.$el ? n == null ? void 0 : n.$el : n;
}, Tr = (e) => Object.assign({ type: "dot" }, e), Da = (e) => Array.isArray(e) ? !!e[0] && !!e[1] : !1, Gt = {
  prop: (e) => `"${e}" prop must be enabled!`,
  dateArr: (e) => `You need to use array as "model-value" binding in order to support "${e}"`
}, Te = (e) => e, ea = (e) => e === 0 ? e : !e || isNaN(+e) ? null : +e, ta = (e) => e === null, Ar = (e) => {
  if (e)
    return [...e.querySelectorAll("input, button, select, textarea, a[href]")][0];
}, _r = (e) => {
  const n = [], a = (t) => t.filter((o) => o);
  for (let t = 0; t < e.length; t += 3) {
    const o = [e[t], e[t + 1], e[t + 2]];
    n.push(a(o));
  }
  return n;
}, Ct = (e, n, a) => {
  const t = a ?? a === 0, o = n ?? n === 0;
  if (!t && !o)
    return !1;
  const l = +a, c = +n;
  return t && o ? +e > l || +e < c : t ? +e > l : o ? +e < c : !1;
}, bt = (e, n) => _r(e).map((a) => a.map((t) => {
  const { active: o, disabled: l, isBetween: c } = n(t);
  return {
    ...t,
    active: o,
    disabled: l,
    className: {
      dp__overlay_cell_active: o,
      dp__overlay_cell: !o,
      dp__overlay_cell_disabled: l,
      dp__overlay_cell_pad: !0,
      dp__overlay_cell_active_disabled: l && o,
      dp__cell_in_between: c
    }
  };
})), na = (e, n, a, t, o) => {
  const l = bn(e, n.slice(0, e.length), /* @__PURE__ */ new Date());
  return zt(l) && Wa(l) ? t || o ? l : De(l, {
    hours: +a.hours,
    minutes: +(a == null ? void 0 : a.minutes),
    seconds: +(a == null ? void 0 : a.seconds),
    milliseconds: 0
  }) : null;
}, Sr = (e, n, a, t, o) => {
  const l = Array.isArray(a) ? a[0] : a;
  if (typeof n == "string")
    return na(e, n, l, t, o);
  if (Array.isArray(n)) {
    let c = null;
    for (const b of n)
      if (c = na(e, b, l, t, o), c)
        break;
    return c;
  }
  return typeof n == "function" ? n(e) : null;
}, S = (e) => e ? new Date(e) : /* @__PURE__ */ new Date(), Pr = (e, n, a) => {
  if (n) {
    const o = (e.getMonth() + 1).toString().padStart(2, "0"), l = e.getDate().toString().padStart(2, "0"), c = e.getHours().toString().padStart(2, "0"), b = e.getMinutes().toString().padStart(2, "0"), p = a ? e.getSeconds().toString().padStart(2, "0") : "00";
    return `${e.getFullYear()}-${o}-${l}T${c}:${b}:${p}.000Z`;
  }
  const t = Date.UTC(
    e.getUTCFullYear(),
    e.getUTCMonth(),
    e.getUTCDate(),
    e.getUTCHours(),
    e.getUTCMinutes(),
    e.getUTCSeconds()
  );
  return new Date(t).toISOString();
}, Le = (e) => {
  let n = S(JSON.parse(JSON.stringify(e)));
  return n = ia(n, 0), n = da(n, 0), n = ca(n, 0), n = Pn(n, 0), n;
}, tt = (e, n, a, t) => {
  let o = e ? S(e) : S();
  return (n || n === 0) && (o = ia(o, +n)), (a || a === 0) && (o = da(o, +a)), (t || t === 0) && (o = ca(o, +t)), Pn(o, 0);
}, Pe = (e, n) => !e || !n ? !1 : _t(Le(e), Le(n)), ye = (e, n) => !e || !n ? !1 : ht(Le(e), Le(n)), Ne = (e, n) => !e || !n ? !1 : St(Le(e), Le(n)), Vn = (e, n, a) => e != null && e[0] && (e != null && e[1]) ? Ne(a, e[0]) && Pe(a, e[1]) : e != null && e[0] && n ? Ne(a, e[0]) && Pe(a, n) || Pe(a, e[0]) && Ne(a, n) : !1, ze = (e) => {
  const n = De(new Date(e), { date: 1 });
  return Le(n);
}, cn = (e, n, a) => n && (a || a === 0) ? Object.fromEntries(
  ["hours", "minutes", "seconds"].map((t) => t === n ? [t, a] : [t, isNaN(+e[t]) ? void 0 : +e[t]])
) : {
  hours: isNaN(+e.hours) ? void 0 : +e.hours,
  minutes: isNaN(+e.minutes) ? void 0 : +e.minutes,
  seconds: isNaN(+e.seconds) ? void 0 : +e.seconds
}, ft = (e) => ({
  hours: qe(e),
  minutes: xe(e),
  seconds: pt(e)
}), Ma = (e, n) => {
  if (n) {
    const a = fe(S(n));
    if (a > e)
      return 12;
    if (a === e)
      return ge(S(n));
  }
}, $a = (e, n) => {
  if (n) {
    const a = fe(S(n));
    return a < e ? -1 : a === e ? ge(S(n)) : void 0;
  }
}, kt = (e) => {
  if (e)
    return fe(S(e));
}, Ze = (e, n) => n ? kr(e, n) : e, Cr = (e, n) => n ? wr(e, n) : e, aa = (e) => e instanceof Date ? e : La(e), Ta = (e, n) => {
  const a = Ne(e, n) ? n : e, t = Ne(n, e) ? n : e;
  return fa({ start: a, end: t });
}, Rr = (e) => {
  const n = ct(e, 1);
  return { month: ge(n), year: fe(n) };
}, jt = (e, n, a) => {
  const t = va(Ze(e, n), { weekStartsOn: +a }), o = Ua(Ze(e, n), { weekStartsOn: +a });
  return [t, o];
}, Aa = (e, n) => {
  const a = {
    hours: qe(S()),
    minutes: xe(S()),
    seconds: n ? pt(S()) : 0
  };
  return Object.assign(a, e);
}, et = (e, n, a) => [De(S(e), { date: 1 }), De(S(), { month: n, year: a, date: 1 })], Je = (e, n, a) => {
  let t = e ? S(e) : S();
  return (n || n === 0) && (t = Ha(t, n)), a && (t = dt(t, a)), t;
}, _a = (e, n, a, t, o) => {
  if (!t || o && !n || !o && !a)
    return !1;
  const l = o ? ct(e, 1) : Pt(e, 1), c = [ge(l), fe(l)];
  return o ? !Or(...c, n) : !Nr(...c, a);
}, Nr = (e, n, a) => Pe(...et(a, e, n)) || ye(...et(a, e, n)), Or = (e, n, a) => Ne(...et(a, e, n)) || ye(...et(a, e, n)), Sa = (e, n, a, t, o, l) => {
  if (typeof n == "function")
    return n(e);
  const c = a ? { locale: a } : void 0;
  return Array.isArray(e) ? `${it(e[0], l, c)}${o && !e[1] ? "" : t}${e[1] ? it(e[1], l, c) : ""}` : it(e, l, c);
}, gt = (e) => {
  if (e)
    return null;
  throw new Error(Gt.prop("partial-range"));
}, Ht = (e, n) => {
  if (n)
    return e();
  throw new Error(Gt.prop("range"));
}, An = (e) => Array.isArray(e) ? zt(e[0]) && (e[1] ? zt(e[1]) : !0) : e ? zt(e) : !1, Ir = (e) => De(S(), {
  hours: +e.hours || 0,
  minutes: +e.minutes || 0,
  seconds: +e.seconds || 0
}), fn = (e, n, a, t) => {
  if (!e)
    return !0;
  if (t) {
    const o = a === "max" ? _t(e, n) : St(e, n), l = { seconds: 0, milliseconds: 0 };
    return o || ht(De(e, l), De(n, l));
  }
  return a === "max" ? e.getTime() <= n.getTime() : e.getTime() >= n.getTime();
}, ra = (e, n, a, t, o) => {
  const l = e ? Ir(e) : S(n);
  return Array.isArray(t) ? fn(t[0], l, a, !!n) && fn(t[1], l, a, !!n) && o : fn(t, l, a, !!n) && o;
}, vn = (e) => De(S(), ft(e)), Yr = (e, n) => Array.isArray(e) ? e.map((a) => S(a)).filter((a) => fe(S(a)) === n).map((a) => ge(a)) : [], Tt = Nt({
  menuFocused: !1,
  shiftKeyInMenu: !1
}), Pa = () => {
  const e = (t) => {
    Tt.menuFocused = t;
  }, n = (t) => {
    Tt.shiftKeyInMenu !== t && (Tt.shiftKeyInMenu = t);
  };
  return {
    control: H(() => ({ shiftKeyInMenu: Tt.shiftKeyInMenu, menuFocused: Tt.menuFocused })),
    setMenuFocused: e,
    setShiftKey: n
  };
}, be = Nt({
  monthYear: [],
  calendar: [],
  time: [],
  actionRow: [],
  selectionGrid: [],
  timePicker: {
    0: [],
    1: []
  },
  monthPicker: []
}), mn = z(null), Lt = z(!1), gn = z(!1), yn = z(!1), hn = z(!1), Oe = z(0), Se = z(0), rt = () => {
  const e = H(() => Lt.value ? [...be.selectionGrid, be.actionRow].filter((y) => y.length) : gn.value ? [
    ...be.timePicker[0],
    ...be.timePicker[1],
    hn.value ? [] : [mn.value],
    be.actionRow
  ].filter((y) => y.length) : yn.value ? [...be.monthPicker, be.actionRow] : [be.monthYear, ...be.calendar, be.time, be.actionRow].filter((y) => y.length)), n = (y) => {
    Oe.value = y ? Oe.value + 1 : Oe.value - 1;
    let N = null;
    e.value[Se.value] && (N = e.value[Se.value][Oe.value]), N || (Oe.value = y ? Oe.value - 1 : Oe.value + 1);
  }, a = (y) => {
    if (Se.value === 0 && !y || Se.value === e.value.length && y)
      return;
    Se.value = y ? Se.value + 1 : Se.value - 1, e.value[Se.value] ? e.value[Se.value] && !e.value[Se.value][Oe.value] && Oe.value !== 0 && (Oe.value = e.value[Se.value].length - 1) : Se.value = y ? Se.value - 1 : Se.value + 1;
  }, t = (y) => {
    let N = null;
    e.value[Se.value] && (N = e.value[Se.value][Oe.value]), N ? N.focus({ preventScroll: !Lt.value }) : Oe.value = y ? Oe.value - 1 : Oe.value + 1;
  }, o = () => {
    n(!0), t(!0);
  }, l = () => {
    n(!1), t(!1);
  }, c = () => {
    a(!1), t(!0);
  }, b = () => {
    a(!0), t(!0);
  }, p = (y, N) => {
    be[N] = y;
  }, M = (y, N) => {
    be[N] = y;
  }, k = () => {
    Oe.value = 0, Se.value = 0;
  };
  return {
    buildMatrix: p,
    buildMultiLevelMatrix: M,
    setTimePickerBackRef: (y) => {
      mn.value = y;
    },
    setSelectionGrid: (y) => {
      Lt.value = y, k(), y || (be.selectionGrid = []);
    },
    setTimePicker: (y, N = !1) => {
      gn.value = y, hn.value = N, k(), y || (be.timePicker[0] = [], be.timePicker[1] = []);
    },
    setTimePickerElements: (y, N = 0) => {
      be.timePicker[N] = y;
    },
    arrowRight: o,
    arrowLeft: l,
    arrowUp: c,
    arrowDown: b,
    clearArrowNav: () => {
      be.monthYear = [], be.calendar = [], be.time = [], be.actionRow = [], be.selectionGrid = [], be.timePicker[0] = [], be.timePicker[1] = [], Lt.value = !1, gn.value = !1, hn.value = !1, yn.value = !1, k(), mn.value = null;
    },
    setMonthPicker: (y) => {
      yn.value = y, k();
    },
    refSets: be
    // exposed for testing
  };
}, la = (e) => ({
  menuAppearTop: "dp-menu-appear-top",
  menuAppearBottom: "dp-menu-appear-bottom",
  open: "dp-slide-down",
  close: "dp-slide-up",
  next: "calendar-next",
  previous: "calendar-prev",
  vNext: "dp-slide-up",
  vPrevious: "dp-slide-down",
  ...e ?? {}
}), Br = (e) => ({
  toggleOverlay: "Toggle overlay",
  menu: "Datepicker menu",
  input: "Datepicker input",
  calendarWrap: "Calendar wrapper",
  calendarDays: "Calendar days",
  openTimePicker: "Open time picker",
  closeTimePicker: "Close time Picker",
  incrementValue: (n) => `Increment ${n}`,
  decrementValue: (n) => `Decrement ${n}`,
  openTpOverlay: (n) => `Open ${n} overlay`,
  amPmButton: "Switch AM/PM mode",
  openYearsOverlay: "Open years overlay",
  openMonthsOverlay: "Open months overlay",
  nextMonth: "Next month",
  prevMonth: "Previous month",
  nextYear: "Next year",
  prevYear: "Previous year",
  day: () => "",
  ...e ?? {}
}), oa = (e) => e ? typeof e == "boolean" ? e ? 2 : 0 : +e >= 2 ? +e : 2 : 0, Er = (e) => {
  const n = typeof e == "object" && e, a = {
    static: !0,
    solo: !1
  };
  if (!e)
    return { ...a, count: oa(!1) };
  const t = n ? e : {}, o = n ? t.count ?? !0 : e, l = oa(o);
  return Object.assign(a, t, { count: l });
}, Fr = (e, n, a) => e || (typeof a == "string" ? a : n), Vr = (e) => typeof e == "boolean" ? e ? la({}) : !1 : la(e), Hr = (e) => {
  const n = {
    enterSubmit: !0,
    tabSubmit: !0,
    openMenu: !0,
    rangeSeparator: " - "
  };
  return typeof e == "object" ? { ...n, ...e ?? {}, enabled: !0 } : { ...n, enabled: e };
}, Lr = (e) => ({
  months: [],
  years: [],
  times: { hours: [], minutes: [], seconds: [] },
  ...e ?? {}
}), Ur = (e) => ({
  showSelect: !0,
  showCancel: !0,
  showNow: !1,
  showPreview: !0,
  ...e ?? {}
}), Wr = (e) => {
  const n = { input: !1 };
  return typeof e == "object" ? { ...n, ...e ?? {}, enabled: !0 } : {
    enabled: e,
    ...n
  };
}, Ce = (e) => {
  const n = () => {
    const j = e.enableSeconds ? ":ss" : "";
    return e.is24 ? `HH:mm${j}` : `hh:mm${j} aa`;
  }, a = () => e.format ? e.format : e.monthPicker ? "MM/yyyy" : e.timePicker ? n() : e.weekPicker ? "MM/dd/yyyy" : e.yearPicker ? "yyyy" : e.enableTimePicker ? `MM/dd/yyyy, ${n()}` : "MM/dd/yyyy", t = (j) => Aa(j, e.enableSeconds), o = () => e.range ? e.startTime && Array.isArray(e.startTime) ? [t(e.startTime[0]), t(e.startTime[1])] : null : e.startTime && !Array.isArray(e.startTime) ? t(e.startTime) : null, l = H(() => Er(e.multiCalendars)), c = H(() => o()), b = H(() => Br(e.ariaLabels)), p = H(() => Lr(e.filters)), M = H(() => Vr(e.transitions)), k = H(() => Ur(e.actionRow)), T = H(
    () => Fr(e.previewFormat, e.format, a())
  ), Y = H(() => Hr(e.textInput)), q = H(() => Wr(e.inline));
  return {
    defaultedTransitions: M,
    defaultedMultiCalendars: l,
    defaultedStartTime: c,
    defaultedAriaLabels: b,
    defaultedFilters: p,
    defaultedActionRow: k,
    defaultedPreviewFormat: T,
    defaultedTextInput: Y,
    defaultedInline: q,
    getDefaultPattern: a,
    getDefaultStartTime: o
  };
}, zr = (e, n, a) => {
  const t = z(), { defaultedTextInput: o, getDefaultPattern: l } = Ce(n), c = z(""), b = Kt(n, "format");
  mt(t, () => {
    e("internal-model-change", t.value);
  }), mt(b, () => {
    E();
  });
  const p = (r) => Cr(r, n.timezone), M = (r) => Ze(r, n.timezone), k = (r, L) => Sa(
    r,
    n.format,
    n.formatLocale,
    o.value.rangeSeparator,
    n.modelAuto,
    L ?? l()
  ), T = (r) => {
    const L = r ?? S();
    return n.modelType ? w(L) : {
      hours: qe(L),
      minutes: xe(L),
      seconds: n.enableSeconds ? pt(L) : 0
    };
  }, Y = (r) => n.modelType ? w(r) : { month: ge(r), year: fe(r) }, q = (r) => Array.isArray(r) ? Ht(
    () => [
      dt(S(), r[0]),
      r[1] ? dt(S(), r[1]) : gt(n.partialRange)
    ],
    n.range
  ) : dt(S(), +r), j = (r, L) => (typeof r == "string" || typeof r == "number") && n.modelType ? d(r) : L, Q = (r) => Array.isArray(r) ? [
    j(
      r[0],
      tt(null, +r[0].hours, +r[0].minutes, r[0].seconds)
    ),
    j(
      r[1],
      tt(null, +r[1].hours, +r[1].minutes, r[1].seconds)
    )
  ] : j(r, tt(null, r.hours, r.minutes, r.seconds)), B = (r) => Array.isArray(r) ? n.multiDates ? r.map((L) => j(L, Je(null, +L.month, +L.year))) : Ht(
    () => [
      j(r[0], Je(null, +r[0].month, +r[0].year)),
      j(
        r[1],
        r[1] ? Je(null, +r[1].month, +r[1].year) : gt(n.partialRange)
      )
    ],
    n.range
  ) : j(r, Je(null, +r.month, +r.year)), y = (r) => {
    if (Array.isArray(r))
      return r.map((L) => d(L));
    throw new Error(Gt.dateArr("multi-dates"));
  }, N = (r) => {
    if (Array.isArray(r))
      return [S(r[0]), S(r[1])];
    throw new Error(Gt.dateArr("week-picker"));
  }, U = (r) => n.modelAuto ? Array.isArray(r) ? [d(r[0]), d(r[1])] : n.autoApply ? [d(r)] : [d(r), null] : Array.isArray(r) ? Ht(
    () => [
      d(r[0]),
      r[1] ? d(r[1]) : gt(n.partialRange)
    ],
    n.range
  ) : d(r), W = () => {
    Array.isArray(t.value) && n.range && t.value.length === 1 && t.value.push(gt(n.partialRange));
  }, A = () => {
    const r = t.value;
    return [
      w(r[0]),
      r[1] ? w(r[1]) : gt(n.partialRange)
    ];
  }, x = () => t.value[1] ? A() : w(Te(t.value[0])), Z = () => (t.value || []).map((r) => w(r)), le = () => (W(), n.modelAuto ? x() : n.multiDates ? Z() : Array.isArray(t.value) ? Ht(() => A(), n.range) : w(Te(t.value))), v = (r) => !r || Array.isArray(r) && !r.length ? null : n.timePicker ? Q(Te(r)) : n.monthPicker ? B(Te(r)) : n.yearPicker ? q(Te(r)) : n.multiDates ? y(Te(r)) : n.weekPicker ? N(Te(r)) : U(Te(r)), D = (r) => {
    const L = v(r);
    An(Te(L)) ? (t.value = Te(L), E()) : (t.value = null, c.value = "");
  }, P = () => {
    const r = (L) => it(L, o.value.format);
    return `${r(t.value[0])} ${o.value.rangeSeparator} ${t.value[1] ? r(t.value[1]) : ""}`;
  }, K = () => a.value && t.value ? Array.isArray(t.value) ? P() : it(t.value, o.value.format) : k(t.value), f = () => t.value ? n.multiDates ? t.value.map((r) => k(r)).join("; ") : o.value.enabled && typeof o.value.format == "string" ? K() : k(t.value) : "", E = () => {
    !n.format || typeof n.format == "string" || o.value.enabled && typeof o.value.format == "string" ? c.value = f() : c.value = n.format(t.value);
  }, d = (r) => {
    if (n.utc) {
      const L = new Date(r);
      return n.utc === "preserve" ? new Date(L.getTime() + L.getTimezoneOffset() * 6e4) : L;
    }
    return n.modelType ? n.modelType === "date" || n.modelType === "timestamp" ? M(new Date(r)) : n.modelType === "format" && (typeof n.format == "string" || !n.format) ? bn(r, l(), /* @__PURE__ */ new Date()) : M(bn(r, n.modelType, /* @__PURE__ */ new Date())) : M(new Date(r));
  }, w = (r) => r ? n.utc ? Pr(r, n.utc === "preserve", n.enableSeconds) : n.modelType ? n.modelType === "timestamp" ? +p(r) : n.modelType === "format" && (typeof n.format == "string" || !n.format) ? k(p(r)) : k(p(r), n.modelType) : p(r) : "", u = (r, L = !1) => {
    if (e("update:model-value", r), n.emitTimezone && L) {
      const C = Array.isArray(r) ? r.map((g) => Ze(Te(g)), n.emitTimezone) : Ze(Te(r), n.emitTimezone);
      e("update:model-timezone-value", C);
    }
  }, h = (r) => Array.isArray(t.value) ? n.multiDates ? t.value.map((L) => r(L)) : [
    r(t.value[0]),
    t.value[1] ? r(t.value[1]) : gt(n.partialRange)
  ] : r(Te(t.value)), s = (r) => u(Te(h(r)));
  return {
    inputValue: c,
    internalModelValue: t,
    checkBeforeEmit: () => t.value ? n.range ? n.partialRange ? t.value.length >= 1 : t.value.length === 2 : !!t.value : !1,
    parseExternalModelValue: D,
    formatInputValue: E,
    emitModelValue: () => (E(), n.monthPicker ? s(Y) : n.timePicker ? s(T) : n.yearPicker ? s(fe) : n.weekPicker ? u(t.value, !0) : u(le(), !0))
  };
}, jr = (e, n) => {
  const { defaultedFilters: a } = Ce(e), { validateMonthYearInRange: t } = Bt(e), o = (M, k) => {
    let T = M;
    return a.value.months.includes(ge(T)) ? (T = k ? ct(M, 1) : Pt(M, 1), o(T, k)) : T;
  }, l = (M, k) => {
    let T = M;
    return a.value.years.includes(fe(T)) ? (T = k ? ma(M, 1) : ga(M, 1), l(T, k)) : T;
  }, c = (M) => {
    const k = De(/* @__PURE__ */ new Date(), { month: e.month, year: e.year });
    let T = M ? ct(k, 1) : Pt(k, 1);
    e.disableYearSelect && (T = dt(T, e.year));
    let Y = ge(T), q = fe(T);
    a.value.months.includes(Y) && (T = o(T, M), Y = ge(T), q = fe(T)), a.value.years.includes(q) && (T = l(T, M), q = fe(T)), t(Y, q, M, e.preventMinMaxNavigation) && b(Y, q);
  }, b = (M, k) => {
    n("update-month-year", { month: M, year: k });
  }, p = H(() => (M) => _a(
    De(/* @__PURE__ */ new Date(), { month: e.month, year: e.year }),
    e.maxDate,
    e.minDate,
    e.preventMinMaxNavigation,
    M
  ));
  return { handleMonthYearChange: c, isDisabled: p, updateMonthYear: b };
};
var yt = /* @__PURE__ */ ((e) => (e.center = "center", e.left = "left", e.right = "right", e))(yt || {}), We = /* @__PURE__ */ ((e) => (e.month = "month", e.year = "year", e))(We || {}), st = /* @__PURE__ */ ((e) => (e.top = "top", e.bottom = "bottom", e))(st || {}), vt = /* @__PURE__ */ ((e) => (e.header = "header", e.calendar = "calendar", e.timePicker = "timePicker", e))(vt || {});
const Kr = (e, n, a, t, o, l, c) => {
  const b = z({}), p = z(!1), M = z({
    top: "0",
    left: "0"
  }), k = z(!1), T = Kt(c, "teleportCenter");
  mt(T, () => {
    M.value = JSON.parse(JSON.stringify({})), U();
  });
  const Y = (d) => {
    if (c.teleport) {
      const w = d.getBoundingClientRect();
      return {
        left: w.left + window.scrollX,
        top: w.top + window.scrollY
      };
    }
    return { top: 0, left: 0 };
  }, q = (d, w) => {
    M.value.left = `${d + w - b.value.width}px`;
  }, j = (d) => {
    M.value.left = `${d}px`;
  }, Q = (d, w) => {
    c.position === yt.left && j(d), c.position === yt.right && q(d, w), c.position === yt.center && (M.value.left = `${d + w / 2 - b.value.width / 2}px`);
  }, B = (d) => {
    const { width: w, height: u } = d.getBoundingClientRect(), { top: h, left: s } = c.altPosition ? c.altPosition(d) : Y(d);
    return { top: +h, left: +s, width: w, height: u };
  }, y = () => {
    M.value.left = "50%", M.value.top = "50%", M.value.transform = "translate(-50%, -50%)", M.value.position = "fixed", delete M.value.opacity;
  }, N = () => {
    const d = Ae(a), { top: w, left: u, transform: h } = c.altPosition(d);
    M.value = { top: `${w}px`, left: `${u}px`, transform: h ?? "" };
  }, U = (d = !0) => {
    var w;
    if (!o.value.enabled) {
      if (T.value)
        return y();
      if (c.altPosition !== null)
        return N();
      if (d) {
        const u = c.teleport ? (w = n.value) == null ? void 0 : w.$el : e.value;
        u && (b.value = u.getBoundingClientRect()), l("recalculate-position");
      }
      return D();
    }
  }, W = ({ inputEl: d, left: w, width: u }) => {
    window.screen.width > 768 && !p.value && Q(w, u), Z(d);
  }, A = (d) => {
    const { top: w, left: u, height: h, width: s } = B(d);
    M.value.top = `${h + w + +c.offset}px`, k.value = !1, p.value || (M.value.left = `${u + s / 2 - b.value.width / 2}px`), W({ inputEl: d, left: u, width: s });
  }, x = (d) => {
    const { top: w, left: u, width: h } = B(d);
    M.value.top = `${w - +c.offset - b.value.height}px`, k.value = !0, W({ inputEl: d, left: u, width: h });
  }, Z = (d) => {
    if (c.autoPosition) {
      const { left: w, width: u } = B(d), { left: h, right: s } = b.value;
      if (!p.value) {
        if (Math.abs(h) !== Math.abs(s)) {
          if (h <= 0)
            return p.value = !0, j(w);
          if (s >= document.documentElement.clientWidth)
            return p.value = !0, q(w, u);
        }
        return Q(w, u);
      }
    }
  }, le = () => {
    const d = Ae(a);
    if (d) {
      const { height: w } = b.value, { top: u, height: h } = d.getBoundingClientRect(), _ = window.innerHeight - u - h, te = u;
      return w <= _ ? st.bottom : w > _ && w <= te ? st.top : _ >= te ? st.bottom : st.top;
    }
    return st.bottom;
  }, v = (d) => le() === st.bottom ? A(d) : x(d), D = () => {
    const d = Ae(a);
    if (d)
      return c.autoPosition ? v(d) : A(d);
  }, P = function(d) {
    if (d) {
      const w = d.scrollHeight > d.clientHeight, h = window.getComputedStyle(d).overflowY.indexOf("hidden") !== -1;
      return w && !h;
    }
    return !0;
  }, K = function(d) {
    return !d || d === document.body || d.nodeType === Node.DOCUMENT_FRAGMENT_NODE ? window : P(d) ? d : K(d.parentNode);
  }, f = (d) => {
    if (d)
      switch (c.position) {
        case yt.left:
          return { left: 0, transform: "translateX(0)" };
        case yt.right:
          return { left: `${d.width}px`, transform: "translateX(-100%)" };
        default:
          return { left: `${d.width / 2}px`, transform: "translateX(-50%)" };
      }
    return {};
  };
  return {
    openOnTop: k,
    menuStyle: M,
    xCorrect: p,
    setMenuPosition: U,
    getScrollableParent: K,
    shadowRender: (d, w) => {
      var L, C, g;
      const u = document.createElement("div"), h = (L = Ae(a)) == null ? void 0 : L.getBoundingClientRect();
      u.setAttribute("id", "dp--temp-container");
      const s = (C = t.value) != null && C.clientWidth ? t.value : document.body;
      s.append(u);
      const _ = document.getElementById("dp--temp-container"), te = f(h), r = Ya(d, {
        ...w,
        shadow: !0,
        style: { opacity: 0, position: "absolute", ...te }
      });
      Wn(r, _), b.value = (g = r.el) == null ? void 0 : g.getBoundingClientRect(), Wn(null, _), s.removeChild(_);
    }
  };
}, ot = [
  { name: "clock-icon", use: ["time", "calendar", "shared"] },
  { name: "arrow-left", use: ["month-year", "calendar", "shared"] },
  { name: "arrow-right", use: ["month-year", "calendar", "shared"] },
  { name: "arrow-up", use: ["time", "calendar", "month-year", "shared"] },
  { name: "arrow-down", use: ["time", "calendar", "month-year", "shared"] },
  { name: "calendar-icon", use: ["month-year", "time", "calendar", "shared"] },
  { name: "day", use: ["calendar", "shared"] },
  { name: "month-overlay-value", use: ["calendar", "month-year", "shared"] },
  { name: "year-overlay-value", use: ["calendar", "month-year", "shared"] },
  { name: "year-overlay", use: ["month-year", "shared"] },
  { name: "month-overlay", use: ["month-year", "shared"] },
  { name: "month-overlay-header", use: ["month-year", "shared"] },
  { name: "year-overlay-header", use: ["month-year", "shared"] },
  { name: "hours-overlay-value", use: ["calendar", "time", "shared"] },
  { name: "minutes-overlay-value", use: ["calendar", "time", "shared"] },
  { name: "seconds-overlay-value", use: ["calendar", "time", "shared"] },
  { name: "hours", use: ["calendar", "time", "shared"] },
  { name: "minutes", use: ["calendar", "time", "shared"] },
  { name: "month", use: ["calendar", "month-year", "shared"] },
  { name: "year", use: ["calendar", "month-year", "shared"] },
  { name: "action-buttons", use: ["action"] },
  { name: "action-preview", use: ["action"] },
  { name: "calendar-header", use: ["calendar", "shared"] },
  { name: "marker-tooltip", use: ["calendar", "shared"] },
  { name: "action-extra", use: ["menu"] },
  { name: "time-picker-overlay", use: ["calendar", "time", "shared"] },
  { name: "am-pm-button", use: ["calendar", "time", "shared"] },
  { name: "left-sidebar", use: ["menu"] },
  { name: "right-sidebar", use: ["menu"] },
  { name: "month-year", use: ["month-year", "shared"] },
  { name: "time-picker", use: ["menu", "shared"] },
  { name: "action-row", use: ["action"] },
  { name: "marker", use: ["calendar", "shared"] }
], Gr = [{ name: "trigger" }, { name: "input-icon" }, { name: "clear-icon" }, { name: "dp-input" }], Zr = {
  all: () => ot,
  monthYear: () => ot.filter((e) => e.use.includes("month-year")),
  input: () => Gr,
  timePicker: () => ot.filter((e) => e.use.includes("time")),
  action: () => ot.filter((e) => e.use.includes("action")),
  calendar: () => ot.filter((e) => e.use.includes("calendar")),
  menu: () => ot.filter((e) => e.use.includes("menu")),
  shared: () => ot.filter((e) => e.use.includes("shared"))
}, je = (e, n, a) => {
  const t = [];
  return Zr[n]().forEach((o) => {
    e[o.name] && t.push(o.name);
  }), a != null && a.length && a.forEach((o) => {
    o.slot && t.push(o.slot);
  }), t;
}, Yt = (e) => {
  const n = H(() => (t) => e.value ? t ? e.value.open : e.value.close : ""), a = H(() => (t) => e.value ? t ? e.value.menuAppearTop : e.value.menuAppearBottom : "");
  return { transitionName: n, showTransition: !!e.value, menuTransition: a };
}, Zt = (e, n) => {
  const a = z([{ month: ge(S()), year: fe(S()) }]), t = Nt({
    hours: e.range ? [qe(S()), qe(S())] : qe(S()),
    minutes: e.range ? [xe(S()), xe(S())] : xe(S()),
    seconds: e.range ? [0, 0] : 0
  }), o = H({
    get: () => e.internalModelValue,
    set: (b) => {
      !e.readonly && !e.disabled && n("update:internal-model-value", b);
    }
  }), l = H(
    () => (b) => a.value[b] ? a.value[b].month : 0
  ), c = H(
    () => (b) => a.value[b] ? a.value[b].year : 0
  );
  return {
    calendars: a,
    time: t,
    modelValue: o,
    month: l,
    year: c
  };
}, qr = (e, n) => {
  const { defaultedMultiCalendars: a } = Ce(n), { isDisabled: t, matchDate: o } = Bt(n), l = z(null), c = z(S()), b = (s) => {
    !s.current && n.hideOffsetDates || (l.value = s.value);
  }, p = () => {
    l.value = null;
  }, M = (s) => Array.isArray(e.value) && n.range && e.value[0] && l.value ? s ? Ne(l.value, e.value[0]) : Pe(l.value, e.value[0]) : !0, k = (s, _) => {
    const te = () => e.value ? _ ? e.value[0] || null : e.value[1] : null, r = e.value && Array.isArray(e.value) ? te() : null;
    return ye(S(s.value), r);
  }, T = (s) => {
    const _ = Array.isArray(e.value) ? e.value[0] : null;
    return s ? !Pe(l.value ?? null, _) : !0;
  }, Y = (s, _ = !0) => (n.range || n.weekPicker) && Array.isArray(e.value) && e.value.length === 2 ? n.hideOffsetDates && !s.current ? !1 : ye(S(s.value), e.value[_ ? 0 : 1]) : n.range ? k(s, _) && T(_) || ye(s.value, Array.isArray(e.value) ? e.value[0] : null) && M(_) : !1, q = (s, _, te) => Array.isArray(e.value) && e.value[0] && e.value.length === 1 ? s ? !1 : te ? Ne(e.value[0], _.value) : Pe(e.value[0], _.value) : !1, j = (s) => !e.value || n.hideOffsetDates && !s.current ? !1 : n.range ? n.modelAuto && Array.isArray(e.value) ? ye(s.value, e.value[0] ? e.value[0] : c.value) : !1 : n.multiDates && Array.isArray(e.value) ? e.value.some((_) => ye(_, s.value)) : ye(s.value, e.value ? e.value : c.value), Q = (s) => {
    if (n.autoRange || n.weekPicker) {
      if (l.value) {
        if (n.hideOffsetDates && !s.current)
          return !1;
        const _ = ut(l.value, +n.autoRange), te = jt(S(l.value), n.timezone, n.weekStart);
        return n.weekPicker ? ye(te[1], S(s.value)) : ye(_, S(s.value));
      }
      return !1;
    }
    return !1;
  }, B = (s) => {
    if (n.autoRange || n.weekPicker) {
      if (l.value) {
        const _ = ut(l.value, +n.autoRange);
        if (n.hideOffsetDates && !s.current)
          return !1;
        const te = jt(S(l.value), n.timezone, n.weekStart);
        return n.weekPicker ? Ne(s.value, te[0]) && Pe(s.value, te[1]) : Ne(s.value, l.value) && Pe(s.value, _);
      }
      return !1;
    }
    return !1;
  }, y = (s) => {
    if (n.autoRange || n.weekPicker) {
      if (l.value) {
        if (n.hideOffsetDates && !s.current)
          return !1;
        const _ = jt(S(l.value), n.timezone, n.weekStart);
        return n.weekPicker ? ye(_[0], s.value) : ye(l.value, s.value);
      }
      return !1;
    }
    return !1;
  }, N = (s) => Vn(e.value, l.value, s.value), U = () => n.modelAuto && Array.isArray(n.internalModelValue) ? !!n.internalModelValue[0] : !1, W = () => n.modelAuto ? Da(n.internalModelValue) : !0, A = (s) => {
    if (Array.isArray(e.value) && e.value.length || n.weekPicker)
      return !1;
    const _ = n.range ? !Y(s) && !Y(s, !1) : !0;
    return !t(s.value) && !j(s) && !(!s.current && n.hideOffsetDates) && _;
  }, x = (s) => n.range ? n.modelAuto ? U() && j(s) : !1 : j(s), Z = (s) => {
    var _;
    return n.highlight ? o(
      s.value,
      (_ = n.arrMapValues) != null && _.highlightedDates ? n.arrMapValues.highlightedDates : n.highlight
    ) : !1;
  }, le = (s) => t(s.value) && n.highlightDisabledDays === !1, v = (s) => {
    var _;
    return (_ = n.highlightWeekDays) == null ? void 0 : _.includes(s.value.getDay());
  }, D = (s) => (n.range || n.weekPicker) && (!(a.value.count > 0) || s.current) && W() && !(!s.current && n.hideOffsetDates) && !j(s) ? N(s) : !1, P = (s) => {
    const { isRangeStart: _, isRangeEnd: te } = E(s), r = n.range ? _ || te : !1;
    return {
      dp__cell_offset: !s.current,
      dp__pointer: !n.disabled && !(!s.current && n.hideOffsetDates) && !t(s.value),
      dp__cell_disabled: t(s.value),
      dp__cell_highlight: !le(s) && (Z(s) || v(s)) && !x(s) && !r,
      dp__cell_highlight_active: !le(s) && (Z(s) || v(s)) && x(s),
      dp__today: !n.noToday && ye(s.value, c.value) && s.current
    };
  }, K = (s) => ({
    dp__active_date: x(s),
    dp__date_hover: A(s)
  }), f = (s) => ({
    ...d(s),
    ...w(s),
    dp__range_between_week: D(s) && n.weekPicker
  }), E = (s) => {
    const _ = a.value.count > 0 ? s.current && Y(s) && W() : Y(s) && W(), te = a.value.count > 0 ? s.current && Y(s, !1) && W() : Y(s, !1) && W();
    return { isRangeStart: _, isRangeEnd: te };
  }, d = (s) => {
    const { isRangeStart: _, isRangeEnd: te } = E(s);
    return {
      dp__range_start: _,
      dp__range_end: te,
      dp__range_between: D(s) && !n.weekPicker,
      dp__date_hover_start: q(A(s), s, !0),
      dp__date_hover_end: q(A(s), s, !1)
    };
  }, w = (s) => ({
    ...d(s),
    dp__cell_auto_range: B(s),
    dp__cell_auto_range_start: y(s),
    dp__cell_auto_range_end: Q(s)
  }), u = (s) => n.range ? n.autoRange ? w(s) : n.modelAuto ? { ...K(s), ...d(s) } : d(s) : n.weekPicker ? f(s) : K(s);
  return {
    setHoverDate: b,
    clearHoverDate: p,
    getDayClassData: (s) => n.hideOffsetDates && !s.current ? {} : {
      ...P(s),
      ...u(s),
      [n.dayClass ? n.dayClass(s.value) : ""]: !0,
      [n.calendarCellClassName]: !!n.calendarCellClassName
    }
  };
}, Bt = (e) => {
  const { defaultedFilters: n } = Ce(e), a = (v) => {
    const D = Le(t(S(v))).toISOString(), [P] = D.split("T");
    return P;
  }, t = (v) => Ze(v, e.timezone), o = (v) => {
    var s;
    const D = e.maxDate ? Ne(t(v), t(S(e.maxDate))) : !1, P = e.minDate ? Pe(t(v), t(S(e.minDate))) : !1, K = p(
      v,
      (s = e.arrMapValues) != null && s.disabledDates ? e.arrMapValues.disabledDates : e.disabledDates
    ), E = n.value.months.map((_) => +_).includes(ge(v)), d = e.disabledWeekDays.length ? e.disabledWeekDays.some((_) => +_ === za(v)) : !1, w = k(v), u = fe(v), h = u < +e.yearRange[0] || u > +e.yearRange[1];
    return !(D || P || K || E || h || d || w);
  }, l = (v, D) => Pe(...et(e.minDate, v, D)) || ye(...et(e.minDate, v, D)), c = (v, D) => Ne(...et(e.maxDate, v, D)) || ye(...et(e.maxDate, v, D)), b = (v, D, P) => {
    let K = !1;
    return e.maxDate && P && c(v, D) && (K = !0), e.minDate && !P && l(v, D) && (K = !0), K;
  }, p = (v, D) => v ? D instanceof Map ? !!D.get(a(v)) : Array.isArray(D) ? D.some((P) => ye(t(S(P)), t(v))) : D ? D(S(JSON.parse(JSON.stringify(v)))) : !1 : !0, M = (v, D, P, K) => {
    let f = !1;
    return K ? e.minDate && e.maxDate ? f = b(v, D, P) : (e.minDate && l(v, D) || e.maxDate && c(v, D)) && (f = !0) : f = !0, f;
  }, k = (v) => {
    var D, P, K, f, E;
    return Array.isArray(e.allowedDates) && !((D = e.allowedDates) != null && D.length) ? !0 : (P = e.arrMapValues) != null && P.allowedDates ? !p(v, (K = e.arrMapValues) == null ? void 0 : K.allowedDates) : (f = e.allowedDates) != null && f.length ? !((E = e.allowedDates) != null && E.some((d) => ye(t(S(d)), t(v)))) : !1;
  }, T = (v) => !o(v), Y = (v) => !fa({ start: v[0], end: v[1] }).some((P) => T(P)), q = (v, D, P = 0) => {
    if (Array.isArray(D) && D[P]) {
      const K = ja(v, D[P]), f = Ta(D[P], v), E = f.length === 1 ? 0 : f.filter((w) => T(w)).length, d = Math.abs(K) - E;
      if (e.minRange && e.maxRange)
        return d >= +e.minRange && d <= +e.maxRange;
      if (e.minRange)
        return d >= +e.minRange;
      if (e.maxRange)
        return d <= +e.maxRange;
    }
    return !0;
  }, j = (v) => new Map(v.map((D) => [a(D), !0])), Q = (v) => Array.isArray(v) && v.length > 0, B = () => {
    const v = {
      disabledDates: null,
      allowedDates: null,
      highlightedDates: null
    };
    return Q(e.allowedDates) && (v.allowedDates = j(e.allowedDates)), Q(e.highlight) && (v.highlightedDates = j(e.highlight)), Q(e.disabledDates) && (v.disabledDates = j(e.disabledDates)), v;
  }, y = () => !e.enableTimePicker || e.monthPicker || e.yearPicker || e.ignoreTimeValidation, N = (v) => Array.isArray(v) ? [v[0] ? vn(v[0]) : null, v[1] ? vn(v[1]) : null] : vn(v), U = (v, D, P) => v.find(
    (K) => +K.hours === qe(D) && K.minutes === "*" ? !0 : +K.minutes === xe(D)
  ) && P, W = (v, D, P) => {
    const [K, f] = v, [E, d] = D;
    return !U(K, E, P) && !U(f, d, P) && P;
  }, A = (v, D) => {
    const P = Array.isArray(D) ? D : [D];
    return Array.isArray(e.disabledTimes) ? Array.isArray(e.disabledTimes[0]) ? W(e.disabledTimes, P, v) : !P.some((K) => U(e.disabledTimes, K, v)) : v;
  }, x = (v, D) => {
    const P = Array.isArray(D) ? [ft(D[0]), D[1] ? ft(D[1]) : void 0] : ft(D), K = !e.disabledTimes(P);
    return v && K;
  }, Z = (v, D) => e.disabledTimes ? Array.isArray(e.disabledTimes) ? A(D, v) : x(D, v) : D;
  return {
    isDisabled: T,
    validateDate: o,
    validateMonthYearInRange: M,
    isDateRangeAllowed: Y,
    checkMinMaxRange: q,
    matchDate: p,
    mapDatesArrToMap: B,
    isValidTime: (v) => {
      let D = !0;
      if (!v || y())
        return !0;
      const P = !e.minDate && !e.maxDate ? N(v) : v;
      return (e.maxTime || e.maxDate) && (D = ra(e.maxTime, e.maxDate, "max", Te(P), D)), (e.minTime || e.minDate) && (D = ra(e.minTime, e.minDate, "min", Te(P), D)), Z(v, D);
    }
  };
}, qt = () => {
  const e = H(() => (t, o) => t == null ? void 0 : t.includes(o)), n = H(() => (t, o) => t.count ? t.solo ? !0 : o === 0 : !0), a = H(() => (t, o) => t.count ? t.solo ? !0 : o === t.count - 1 : !0);
  return { hideNavigationButtons: e, showLeftIcon: n, showRightIcon: a };
}, xr = (e, n, a) => {
  const t = z(0), o = Nt({
    // monthYearInput: !!props.timePicker,
    [vt.timePicker]: !e.enableTimePicker || e.timePicker || e.monthPicker,
    [vt.calendar]: !1,
    [vt.header]: !1
  }), l = (k) => {
    var T;
    (T = e.flow) != null && T.length && (o[k] = !0, Object.keys(o).filter((Y) => !o[Y]).length || M());
  }, c = () => {
    var k;
    (k = e.flow) != null && k.length && t.value !== -1 && (t.value += 1, n("flow-step", t.value), M());
  }, b = () => {
    t.value = -1;
  }, p = (k, T, ...Y) => {
    e.flow[t.value] === k && a.value && a.value[T](...Y);
  }, M = () => {
    p("month", "toggleMonthPicker", !0), p("year", "toggleYearPicker", !0), p("calendar", "toggleTimePicker", !1, !0), p("time", "toggleTimePicker", !0, !0);
    const k = e.flow[t.value];
    (k === "hours" || k === "minutes" || k === "seconds") && p(k, "toggleTimePicker", !0, !0, k);
  };
  return { childMount: l, updateFlowStep: c, resetFlow: b, flowStep: t };
}, xt = {
  multiCalendars: { type: [Boolean, Number, String, Object], default: void 0 },
  modelValue: { type: [String, Date, Array, Object, Number], default: null },
  modelType: { type: String, default: null },
  position: { type: String, default: "center" },
  dark: { type: Boolean, default: !1 },
  format: {
    type: [String, Function],
    default: () => null
  },
  closeOnScroll: { type: Boolean, default: !1 },
  autoPosition: { type: Boolean, default: !0 },
  closeOnAutoApply: { type: Boolean, default: !0 },
  altPosition: { type: Function, default: null },
  transitions: { type: [Boolean, Object], default: !0 },
  formatLocale: { type: Object, default: null },
  utc: { type: [Boolean, String], default: !1 },
  ariaLabels: { type: Object, default: () => ({}) },
  offset: { type: [Number, String], default: 10 },
  hideNavigation: { type: Array, default: () => [] },
  timezone: { type: String, default: null },
  emitTimezone: { type: String, default: null },
  vertical: { type: Boolean, default: !1 },
  disableMonthYearSelect: { type: Boolean, default: !1 },
  disableYearSelect: { type: Boolean, default: !1 },
  menuClassName: { type: String, default: null },
  dayClass: { type: Function, default: null },
  yearRange: { type: Array, default: () => [1900, 2100] },
  calendarCellClassName: { type: String, default: null },
  enableTimePicker: { type: Boolean, default: !0 },
  autoApply: { type: Boolean, default: !1 },
  disabledDates: { type: [Array, Function], default: () => [] },
  monthNameFormat: { type: String, default: "short" },
  startDate: { type: [Date, String], default: null },
  startTime: { type: [Object, Array], default: null },
  hideOffsetDates: { type: Boolean, default: !1 },
  autoRange: { type: [Number, String], default: null },
  noToday: { type: Boolean, default: !1 },
  disabledWeekDays: { type: Array, default: () => [] },
  allowedDates: { type: Array, default: null },
  nowButtonLabel: { type: String, default: "Now" },
  markers: { type: Array, default: () => [] },
  modeHeight: { type: [Number, String], default: 255 },
  escClose: { type: Boolean, default: !0 },
  spaceConfirm: { type: Boolean, default: !0 },
  monthChangeOnArrows: { type: Boolean, default: !0 },
  presetDates: { type: Array, default: () => [] },
  flow: { type: Array, default: () => [] },
  partialFlow: { type: Boolean, default: !1 },
  preventMinMaxNavigation: { type: Boolean, default: !1 },
  minRange: { type: [Number, String], default: null },
  maxRange: { type: [Number, String], default: null },
  multiDatesLimit: { type: [Number, String], default: null },
  reverseYears: { type: Boolean, default: !1 },
  keepActionRow: { type: Boolean, default: !1 },
  weekPicker: { type: Boolean, default: !1 },
  filters: { type: Object, default: () => ({}) },
  arrowNavigation: { type: Boolean, default: !1 },
  disableTimeRangeValidation: { type: Boolean, default: !1 },
  highlight: {
    type: [Array, Function],
    default: null
  },
  highlightWeekDays: {
    type: Array,
    default: null
  },
  highlightDisabledDays: { type: Boolean, default: !1 },
  teleport: { type: [String, Boolean], default: null },
  teleportCenter: { type: Boolean, default: !1 },
  locale: { type: String, default: "en-Us" },
  weekNumName: { type: String, default: "W" },
  weekStart: { type: [Number, String], default: 1 },
  weekNumbers: {
    type: [String, Function],
    default: null
  },
  calendarClassName: { type: String, default: null },
  noSwipe: { type: Boolean, default: !1 },
  monthChangeOnScroll: { type: [Boolean, String], default: !0 },
  dayNames: {
    type: [Function, Array],
    default: null
  },
  monthPicker: { type: Boolean, default: !1 },
  customProps: { type: Object, default: null },
  yearPicker: { type: Boolean, default: !1 },
  modelAuto: { type: Boolean, default: !1 },
  selectText: { type: String, default: "Select" },
  cancelText: { type: String, default: "Cancel" },
  previewFormat: {
    type: [String, Function],
    default: () => ""
  },
  multiDates: { type: Boolean, default: !1 },
  partialRange: { type: Boolean, default: !0 },
  ignoreTimeValidation: { type: Boolean, default: !1 },
  minDate: { type: [Date, String], default: null },
  maxDate: { type: [Date, String], default: null },
  minTime: { type: Object, default: null },
  maxTime: { type: Object, default: null },
  name: { type: String, default: null },
  placeholder: { type: String, default: "" },
  hideInputIcon: { type: Boolean, default: !1 },
  clearable: { type: Boolean, default: !0 },
  state: { type: Boolean, default: null },
  required: { type: Boolean, default: !1 },
  autocomplete: { type: String, default: "off" },
  inputClassName: { type: String, default: null },
  fixedStart: { type: Boolean, default: !1 },
  fixedEnd: { type: Boolean, default: !1 },
  timePicker: { type: Boolean, default: !1 },
  enableSeconds: { type: Boolean, default: !1 },
  is24: { type: Boolean, default: !0 },
  noHoursOverlay: { type: Boolean, default: !1 },
  noMinutesOverlay: { type: Boolean, default: !1 },
  noSecondsOverlay: { type: Boolean, default: !1 },
  hoursGridIncrement: { type: [String, Number], default: 1 },
  minutesGridIncrement: { type: [String, Number], default: 5 },
  secondsGridIncrement: { type: [String, Number], default: 5 },
  hoursIncrement: { type: [Number, String], default: 1 },
  minutesIncrement: { type: [Number, String], default: 1 },
  secondsIncrement: { type: [Number, String], default: 1 },
  range: { type: Boolean, default: !1 },
  uid: { type: String, default: null },
  disabled: { type: Boolean, default: !1 },
  readonly: { type: Boolean, default: !1 },
  inline: { type: [Boolean, Object], default: !1 },
  textInput: { type: [Boolean, Object], default: !1 },
  onClickOutside: { type: Function, default: null },
  noDisabledRange: { type: Boolean, default: !1 },
  sixWeeks: { type: [Boolean, String], default: !1 },
  actionRow: { type: Object, default: () => ({}) },
  allowPreventDefault: { type: Boolean, default: !1 },
  closeOnClearValue: { type: Boolean, default: !0 },
  focusStartDate: { type: Boolean, default: !1 },
  disabledTimes: { type: [Function, Array], default: void 0 },
  showLastInRange: { type: Boolean, default: !0 },
  timePickerInline: { type: Boolean, default: !1 },
  calendar: { type: Function, default: null }
}, Xe = {
  ...xt,
  shadow: { type: Boolean, default: !1 },
  flowStep: { type: Number, default: 0 },
  internalModelValue: { type: [Date, Array], default: null },
  arrMapValues: { type: Object, default: () => ({}) }
}, Jr = {
  key: 1,
  class: "dp__input_wrap"
}, Xr = ["id", "name", "inputmode", "placeholder", "disabled", "readonly", "required", "value", "autocomplete", "aria-label", "onKeydown"], Qr = {
  key: 2,
  class: "dp__clear_icon"
}, el = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "DatepickerInput",
  props: {
    isMenuOpen: { type: Boolean, default: !1 },
    inputValue: { type: String, default: "" },
    ...xt
  },
  emits: [
    "clear",
    "open",
    "update:input-value",
    "set-input-date",
    "close",
    "select-date",
    "set-empty-date",
    "toggle",
    "focus-prev",
    "focus",
    "blur",
    "real-blur"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, { defaultedTextInput: o, defaultedAriaLabels: l, defaultedInline: c, getDefaultPattern: b, getDefaultStartTime: p } = Ce(t), M = z(), k = z(null), T = z(!1), Y = z(!1), q = H(
      () => ({
        dp__pointer: !t.disabled && !t.readonly && !o.value.enabled,
        dp__disabled: t.disabled,
        dp__input_readonly: !o.value.enabled,
        dp__input: !0,
        dp__input_icon_pad: !t.hideInputIcon,
        dp__input_valid: t.state,
        dp__input_invalid: t.state === !1,
        dp__input_focus: T.value || t.isMenuOpen,
        dp__input_reg: !o.value.enabled,
        [t.inputClassName]: !!t.inputClassName
      })
    ), j = () => {
      a("set-input-date", null), t.autoApply && (a("set-empty-date"), M.value = null);
    }, Q = (f) => {
      const E = p();
      return Sr(
        f,
        o.value.format ?? b(),
        E ?? Aa({}, t.enableSeconds),
        t.inputValue,
        Y.value
      );
    }, B = (f) => {
      const { rangeSeparator: E } = o.value, [d, w] = f.split(`${E}`);
      if (d) {
        const u = Q(d.trim()), h = w ? Q(w.trim()) : null, s = u && h ? [u, h] : [u];
        M.value = u ? s : null;
      }
    }, y = () => {
      Y.value = !0;
    }, N = (f) => {
      if (t.range)
        B(f);
      else if (t.multiDates) {
        const E = f.split(";");
        M.value = E.map((d) => Q(d.trim())).filter((d) => d);
      } else
        M.value = Q(f);
    }, U = (f) => {
      var d;
      const E = typeof f == "string" ? f : (d = f.target) == null ? void 0 : d.value;
      E !== "" ? (o.value.openMenu && !t.isMenuOpen && a("open"), N(E), a("set-input-date", M.value)) : j(), Y.value = !1, a("update:input-value", E);
    }, W = (f) => {
      o.value.enabled ? (N(f.target.value), o.value.enterSubmit && An(M.value) && t.inputValue !== "" ? (a("set-input-date", M.value, !0), M.value = null) : o.value.enterSubmit && t.inputValue === "" && (M.value = null, a("clear"))) : Z(f);
    }, A = (f) => {
      o.value.enabled && o.value.tabSubmit && N(f.target.value), o.value.tabSubmit && An(M.value) && t.inputValue !== "" ? (a("set-input-date", M.value, !0), M.value = null) : o.value.tabSubmit && t.inputValue === "" && (M.value = null, a("clear"));
    }, x = () => {
      T.value = !0, a("focus");
    }, Z = (f) => {
      f.preventDefault(), f.stopImmediatePropagation(), f.stopPropagation(), o.value.enabled && o.value.openMenu && !c.value.input && !t.isMenuOpen ? a("open") : o.value.enabled || a("toggle");
    }, le = () => {
      a("real-blur"), T.value = !1, (!t.isMenuOpen || c.value.enabled && c.value.input) && a("blur"), t.autoApply && o.value.enabled && M.value && !t.isMenuOpen && (a("set-input-date", M.value), a("select-date"), M.value = null);
    }, v = () => {
      a("clear");
    }, D = (f) => {
      if (!o.value.enabled) {
        if (f.code === "Tab")
          return;
        f.preventDefault();
      }
    };
    return n({
      focusInput: () => {
        var f;
        (f = k.value) == null || f.focus({ preventScroll: !0 });
      },
      setParsedDate: (f) => {
        M.value = f;
      }
    }), (f, E) => {
      var d;
      return $(), O("div", { onClick: Z }, [
        f.$slots.trigger && !f.$slots["dp-input"] && !m(c).enabled ? ee(f.$slots, "trigger", { key: 0 }) : I("", !0),
        !f.$slots.trigger && (!m(c).enabled || m(c).input) ? ($(), O("div", Jr, [
          f.$slots["dp-input"] && !f.$slots.trigger && !m(c).enabled ? ee(f.$slots, "dp-input", {
            key: 0,
            value: e.inputValue,
            isMenuOpen: e.isMenuOpen,
            onInput: U,
            onEnter: W,
            onTab: A,
            onClear: v,
            onBlur: le,
            onKeypress: D,
            onPaste: y
          }) : I("", !0),
          f.$slots["dp-input"] ? I("", !0) : ($(), O("input", {
            key: 1,
            ref_key: "inputRef",
            ref: k,
            id: f.uid ? `dp-input-${f.uid}` : void 0,
            name: f.name,
            class: he(q.value),
            inputmode: m(o).enabled ? "text" : "none",
            placeholder: f.placeholder,
            disabled: f.disabled,
            readonly: f.readonly,
            required: f.required,
            value: e.inputValue,
            autocomplete: f.autocomplete,
            "aria-label": (d = m(l)) == null ? void 0 : d.input,
            onInput: U,
            onKeydown: [
              oe(W, ["enter"]),
              oe(A, ["tab"]),
              D
            ],
            onBlur: le,
            onFocus: x,
            onKeypress: D,
            onPaste: y
          }, null, 42, Xr)),
          ue("div", {
            onClick: E[2] || (E[2] = (w) => a("toggle"))
          }, [
            f.$slots["input-icon"] && !f.hideInputIcon ? ($(), O("span", {
              key: 0,
              class: "dp__input_icon",
              onClick: E[0] || (E[0] = (w) => a("toggle"))
            }, [
              ee(f.$slots, "input-icon")
            ])) : I("", !0),
            !f.$slots["input-icon"] && !f.hideInputIcon && !f.$slots["dp-input"] ? ($(), ve(m(It), {
              key: 1,
              onClick: E[1] || (E[1] = (w) => a("toggle")),
              class: "dp__input_icon dp__input_icons"
            })) : I("", !0)
          ]),
          f.$slots["clear-icon"] && e.inputValue && f.clearable && !f.disabled && !f.readonly ? ($(), O("span", Qr, [
            ee(f.$slots, "clear-icon", { clear: v })
          ])) : I("", !0),
          f.clearable && !f.$slots["clear-icon"] && e.inputValue && !f.disabled && !f.readonly ? ($(), ve(m(ha), {
            key: 3,
            class: "dp__clear_icon dp__input_icons",
            onClick: pe(v, ["stop", "prevent"])
          }, null, 8, ["onClick"])) : I("", !0)
        ])) : I("", !0)
      ]);
    };
  }
}), tl = ["title"], nl = { class: "dp__action_buttons" }, al = ["onKeydown", "disabled"], rl = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "ActionRow",
  props: {
    menuMount: { type: Boolean, default: !1 },
    calendarWidth: { type: Number, default: 0 },
    ...Xe
  },
  emits: ["close-picker", "select-date", "select-now", "invalid-select"],
  setup(e, { emit: n }) {
    const a = e, {
      defaultedActionRow: t,
      defaultedPreviewFormat: o,
      defaultedMultiCalendars: l,
      defaultedTextInput: c,
      defaultedInline: b,
      getDefaultPattern: p
    } = Ce(a), { isValidTime: M } = Bt(a), { buildMatrix: k } = rt(), T = z(null), Y = z(null);
    Ue(() => {
      a.arrowNavigation && k([Ae(T), Ae(Y)], "actionRow");
    });
    const q = H(() => a.range && !a.partialRange && a.internalModelValue ? a.internalModelValue.length === 2 : !0), j = H(() => !Q.value || !B.value || !q.value), Q = H(() => !a.enableTimePicker || a.ignoreTimeValidation ? !0 : M(a.internalModelValue)), B = H(() => a.monthPicker ? a.range && Array.isArray(a.internalModelValue) ? !a.internalModelValue.filter((D) => !Z(D)).length : Z(a.internalModelValue) : !0), y = () => {
      const v = o.value;
      return a.timePicker || a.monthPicker, v(Te(a.internalModelValue));
    }, N = () => {
      const v = a.internalModelValue;
      return l.value.count > 0 ? `${U(v[0])} - ${U(v[1])}` : [U(v[0]), U(v[1])];
    }, U = (v) => Sa(
      v,
      o.value,
      a.formatLocale,
      c.value.rangeSeparator,
      a.modelAuto,
      p()
    ), W = H(() => !a.internalModelValue || !a.menuMount ? "" : typeof o.value == "string" ? Array.isArray(a.internalModelValue) ? a.internalModelValue.length === 2 && a.internalModelValue[1] ? N() : a.multiDates ? a.internalModelValue.map((v) => `${U(v)}`) : a.modelAuto ? `${U(a.internalModelValue[0])}` : `${U(a.internalModelValue[0])} -` : U(a.internalModelValue) : y()), A = () => a.multiDates ? "; " : " - ", x = H(
      () => Array.isArray(W.value) ? W.value.join(A()) : W.value
    ), Z = (v) => {
      if (!a.monthPicker)
        return !0;
      let D = !0;
      const P = S(ze(v));
      if (a.minDate && a.maxDate) {
        const K = S(ze(a.minDate)), f = S(ze(a.maxDate));
        return Ne(P, K) && Pe(P, f) || ye(P, K) || ye(P, f);
      }
      if (a.minDate) {
        const K = S(ze(a.minDate));
        D = Ne(P, K) || ye(P, K);
      }
      if (a.maxDate) {
        const K = S(ze(a.maxDate));
        D = Pe(P, K) || ye(P, K);
      }
      return D;
    }, le = () => {
      Q.value && B.value && q.value ? n("select-date") : n("invalid-select");
    };
    return (v, D) => ($(), O("div", {
      class: "dp__action_row",
      style: Ge(e.calendarWidth ? { width: `${e.calendarWidth}px` } : {})
    }, [
      v.$slots["action-row"] ? ee(v.$slots, "action-row", Re(Ee({ key: 0 }, {
        internalModelValue: v.internalModelValue,
        disabled: j.value,
        selectDate: () => v.$emit("select-date"),
        closePicker: () => v.$emit("close-picker")
      }))) : ($(), O(ce, { key: 1 }, [
        m(t).showPreview ? ($(), O("div", {
          key: 0,
          class: "dp__selection_preview",
          title: x.value
        }, [
          v.$slots["action-preview"] ? ee(v.$slots, "action-preview", {
            key: 0,
            value: v.internalModelValue
          }) : I("", !0),
          v.$slots["action-preview"] ? I("", !0) : ($(), O(ce, { key: 1 }, [
            nt(Ie(x.value), 1)
          ], 64))
        ], 8, tl)) : I("", !0),
        ue("div", nl, [
          v.$slots["action-buttons"] ? ee(v.$slots, "action-buttons", {
            key: 0,
            value: v.internalModelValue
          }) : I("", !0),
          v.$slots["action-buttons"] ? I("", !0) : ($(), O(ce, { key: 1 }, [
            !m(b).enabled && m(t).showCancel ? ($(), O("button", {
              key: 0,
              type: "button",
              ref_key: "cancelButtonRef",
              ref: T,
              class: "dp__action_button dp__action_cancel",
              onClick: D[0] || (D[0] = (P) => v.$emit("close-picker")),
              onKeydown: [
                D[1] || (D[1] = oe((P) => v.$emit("close-picker"), ["enter"])),
                D[2] || (D[2] = oe((P) => v.$emit("close-picker"), ["space"]))
              ]
            }, Ie(v.cancelText), 545)) : I("", !0),
            m(t).showNow ? ($(), O("button", {
              key: 1,
              type: "button",
              ref_key: "cancelButtonRef",
              ref: T,
              class: "dp__action_button dp__action_cancel",
              onClick: D[3] || (D[3] = (P) => v.$emit("select-now")),
              onKeydown: [
                D[4] || (D[4] = oe((P) => v.$emit("select-now"), ["enter"])),
                D[5] || (D[5] = oe((P) => v.$emit("select-now"), ["space"]))
              ]
            }, Ie(v.nowButtonLabel), 545)) : I("", !0),
            m(t).showSelect ? ($(), O("button", {
              key: 2,
              type: "button",
              class: "dp__action_button dp__action_select",
              onKeydown: [
                oe(le, ["enter"]),
                oe(le, ["space"])
              ],
              onClick: le,
              disabled: j.value,
              ref_key: "selectButtonRef",
              ref: Y
            }, Ie(v.selectText), 41, al)) : I("", !0)
          ], 64))
        ])
      ], 64))
    ], 4));
  }
}), ll = ["onKeydown"], ol = { class: "dp__selection_grid_header" }, sl = ["aria-selected", "aria-disabled", "onClick", "onKeydown", "onMouseover"], ul = ["aria-label", "onKeydown"], Rt = /* @__PURE__ */ Be({
  __name: "SelectionOverlay",
  props: {
    items: {},
    type: {},
    isLast: { type: Boolean },
    arrowNavigation: { type: Boolean },
    skipButtonRef: { type: Boolean },
    headerRefs: {},
    hideNavigation: {},
    escClose: { type: Boolean },
    useRelative: { type: Boolean },
    height: {},
    textInput: { type: [Boolean, Object] }
  },
  emits: ["selected", "toggle", "reset-flow", "hover-value"],
  setup(e, { expose: n, emit: a }) {
    const t = e, { setSelectionGrid: o, buildMultiLevelMatrix: l, setMonthPicker: c } = rt(), { defaultedAriaLabels: b, defaultedTextInput: p } = Ce(t), { hideNavigationButtons: M } = qt(), k = z(!1), T = z(null), Y = z(null), q = z([]), j = z(), Q = z(null), B = z(0), y = z(null);
    Ba(() => {
      T.value = null;
    }), Ue(() => {
      at().then(() => v()), U(), N(!0);
    }), _n(() => N(!1));
    const N = (u) => {
      var h;
      t.arrowNavigation && ((h = t.headerRefs) != null && h.length ? c(u) : o(u));
    }, U = () => {
      var h;
      const u = Ae(Y);
      u && (p.value.enabled || (T.value ? (h = T.value) == null || h.focus({ preventScroll: !0 }) : u.focus({ preventScroll: !0 })), k.value = u.clientHeight < u.scrollHeight);
    }, W = H(
      () => ({
        dp__overlay: !0,
        "dp--overlay-absolute": !t.useRelative,
        "dp--overlay-relative": t.useRelative
      })
    ), A = H(
      () => t.useRelative ? { height: `${t.height}px`, width: "260px" } : void 0
    ), x = H(() => ({
      dp__overlay_col: !0
    })), Z = H(
      () => ({
        dp__btn: !0,
        dp__button: !0,
        dp__overlay_action: !0,
        dp__over_action_scroll: k.value,
        dp__button_bottom: t.isLast
      })
    ), le = H(() => {
      var u, h;
      return {
        dp__overlay_container: !0,
        dp__container_flex: ((u = t.items) == null ? void 0 : u.length) <= 6,
        dp__container_block: ((h = t.items) == null ? void 0 : h.length) > 6
      };
    }), v = () => {
      at().then(() => {
        const u = Ae(T), h = Ae(Y), s = Ae(Q), _ = Ae(y), te = s ? s.getBoundingClientRect().height : 0;
        h && (B.value = h.getBoundingClientRect().height - te), u && _ && (_.scrollTop = u.offsetTop - _.offsetTop - (B.value / 2 - u.getBoundingClientRect().height) - te);
      });
    }, D = (u) => {
      u.disabled || a("selected", u.value);
    }, P = () => {
      a("toggle"), a("reset-flow");
    }, K = () => {
      t.escClose && P();
    }, f = (u, h, s, _) => {
      u && (h.active && (T.value = u), t.arrowNavigation && (Array.isArray(q.value[s]) ? q.value[s][_] = u : q.value[s] = [u], E()));
    }, E = () => {
      var h, s;
      const u = (h = t.headerRefs) != null && h.length ? [t.headerRefs].concat(q.value) : q.value.concat([t.skipButtonRef ? [] : [Q.value]]);
      l(Te(u), (s = t.headerRefs) != null && s.length ? "monthPicker" : "selectionGrid");
    }, d = (u) => {
      t.arrowNavigation || u.stopImmediatePropagation();
    }, w = (u) => {
      j.value = u, a("hover-value", u);
    };
    return n({ focusGrid: U }), (u, h) => {
      var s;
      return $(), O("div", {
        ref_key: "gridWrapRef",
        ref: Y,
        class: he(W.value),
        style: Ge(A.value),
        role: "dialog",
        tabindex: "0",
        onKeydown: [
          oe(pe(K, ["prevent"]), ["esc"]),
          h[0] || (h[0] = oe(pe((_) => d(_), ["prevent"]), ["left"])),
          h[1] || (h[1] = oe(pe((_) => d(_), ["prevent"]), ["up"])),
          h[2] || (h[2] = oe(pe((_) => d(_), ["prevent"]), ["down"])),
          h[3] || (h[3] = oe(pe((_) => d(_), ["prevent"]), ["right"]))
        ]
      }, [
        ue("div", {
          class: he(le.value),
          ref_key: "containerRef",
          ref: y,
          role: "grid",
          style: Ge({ height: `${B.value}px` })
        }, [
          ue("div", ol, [
            ee(u.$slots, "header")
          ]),
          u.$slots.overlay ? ee(u.$slots, "overlay", { key: 0 }) : ($(!0), O(ce, { key: 1 }, Me(u.items, (_, te) => ($(), O("div", {
            class: he(["dp__overlay_row", { dp__flex_row: u.items.length >= 3 }]),
            key: te,
            role: "row"
          }, [
            ($(!0), O(ce, null, Me(_, (r, L) => ($(), O("div", {
              role: "gridcell",
              class: he(x.value),
              key: r.value,
              "aria-selected": r.active,
              "aria-disabled": r.disabled || void 0,
              ref_for: !0,
              ref: (C) => f(C, r, te, L),
              tabindex: "0",
              onClick: (C) => D(r),
              onKeydown: [
                oe(pe((C) => D(r), ["prevent"]), ["enter"]),
                oe(pe((C) => D(r), ["prevent"]), ["space"])
              ],
              onMouseover: (C) => w(r.value)
            }, [
              ue("div", {
                class: he(r.className)
              }, [
                u.$slots.item ? ee(u.$slots, "item", {
                  key: 0,
                  item: r
                }) : I("", !0),
                u.$slots.item ? I("", !0) : ($(), O(ce, { key: 1 }, [
                  nt(Ie(r.text), 1)
                ], 64))
              ], 2)
            ], 42, sl))), 128))
          ], 2))), 128))
        ], 6),
        u.$slots["button-icon"] ? Ut(($(), O("button", {
          key: 0,
          role: "button",
          "aria-label": (s = m(b)) == null ? void 0 : s.toggleOverlay,
          class: he(Z.value),
          tabindex: "0",
          ref_key: "toggleButton",
          ref: Q,
          onClick: P,
          onKeydown: [
            oe(P, ["enter"]),
            oe(P, ["tab"])
          ]
        }, [
          ee(u.$slots, "button-icon")
        ], 42, ul)), [
          [Wt, !m(M)(u.hideNavigation, u.type)]
        ]) : I("", !0)
      ], 46, ll);
    };
  }
}), Hn = /* @__PURE__ */ Be({
  __name: "InstanceWrap",
  props: {
    multiCalendars: {},
    stretch: { type: Boolean }
  },
  setup(e) {
    const n = e, a = H(
      () => n.multiCalendars > 0 ? [...Array(n.multiCalendars).keys()] : [0]
    ), t = H(() => ({
      dp__instance_calendar: n.multiCalendars > 0
    }));
    return (o, l) => ($(), O("div", {
      class: he({
        dp__menu_inner: !o.stretch,
        "dp--menu--inner-stretched": o.stretch,
        dp__flex_display: o.multiCalendars > 0
      })
    }, [
      ($(!0), O(ce, null, Me(a.value, (c, b) => ($(), O("div", {
        key: c,
        class: he(t.value)
      }, [
        ee(o.$slots, "default", {
          instance: c,
          index: b
        })
      ], 2))), 128))
    ], 2));
  }
}), il = ["aria-label", "aria-disabled"], At = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "ArrowBtn",
  props: {
    ariaLabel: {},
    disabled: { type: Boolean }
  },
  emits: ["activate", "set-ref"],
  setup(e, { emit: n }) {
    const a = z(null);
    return Ue(() => n("set-ref", a)), (t, o) => ($(), O("button", {
      type: "button",
      class: "dp__btn dp--arrow-btn-nav",
      onClick: o[0] || (o[0] = (l) => t.$emit("activate")),
      onKeydown: [
        o[1] || (o[1] = oe(pe((l) => t.$emit("activate"), ["prevent"]), ["enter"])),
        o[2] || (o[2] = oe(pe((l) => t.$emit("activate"), ["prevent"]), ["space"]))
      ],
      tabindex: "0",
      "aria-label": t.ariaLabel,
      "aria-disabled": t.disabled || void 0,
      ref_key: "elRef",
      ref: a
    }, [
      ue("span", {
        class: he(["dp__inner_nav", { dp__inner_nav_disabled: t.disabled }])
      }, [
        ee(t.$slots, "default")
      ], 2)
    ], 40, il));
  }
}), Ln = (e, n, a) => {
  if (n.value && Array.isArray(n.value))
    if (n.value.some((t) => ye(e, t))) {
      const t = n.value.filter((o) => !ye(o, e));
      n.value = t.length ? t : null;
    } else
      (a && +a > n.value.length || !a) && n.value.push(e);
  else
    n.value = [e];
}, Ca = (e, n, a) => {
  let t = e.value ? e.value.slice() : [];
  return t.length === 2 && t[1] !== null && (t = []), t.length ? Pe(n, t[0]) ? (t.unshift(n), a("range-start", t[0]), a("range-start", t[1])) : (t[1] = n, a("range-end", n)) : (t = [n], a("range-start", n)), e.value = t, t;
}, Un = (e, n, a, t) => {
  e[0] && e[1] && a && n("auto-apply"), e[0] && !e[1] && t && a && n("auto-apply");
}, dl = (e, n) => {
  const { defaultedMultiCalendars: a, defaultedAriaLabels: t, defaultedTransitions: o } = Ce(e), { modelValue: l, year: c, month: b, calendars: p } = Zt(e, n), M = H(() => wa(e.formatLocale, e.locale, e.monthNameFormat)), k = H(() => Fn(e.yearRange, e.reverseYears)), T = z(null), Y = () => {
    for (let d = 0; d < a.value.count; d++)
      if (d === 0)
        p.value[d] = p.value[0];
      else {
        const w = De(S(), p.value[d - 1]);
        p.value[d] = { month: ge(w), year: fe(ma(w, d)) };
      }
  }, q = (d) => {
    if (!d)
      return Y();
    const w = De(S(), p.value[d]);
    return p.value[0].year = fe(ga(w, a.value.count - 1)), Y();
  }, j = (d) => e.focusStartDate ? d[0] : d[1] ? d[1] : d[0], Q = () => {
    if (l.value) {
      const d = Array.isArray(l.value) ? j(l.value) : l.value;
      p.value[0] = { month: ge(d), year: fe(d) };
    }
  };
  Ue(() => {
    Q(), a.value.count && Y();
  });
  const B = H(() => (d, w) => {
    const u = De(ze(/* @__PURE__ */ new Date()), {
      month: b.value(d),
      year: c.value(d)
    });
    return _a(u, e.maxDate, e.minDate, e.preventMinMaxNavigation, w);
  }), y = (d) => d ? { month: ge(d), year: fe(d) } : { month: null, year: null }, N = () => l.value ? Array.isArray(l.value) ? l.value.map((d) => y(d)) : y(l.value) : y(), U = (d, w) => {
    const u = p.value[d], h = N();
    return Array.isArray(h) ? h.some((s) => s.year === (u == null ? void 0 : u.year) && s.month === w) : (u == null ? void 0 : u.year) === h.year && w === h.month;
  }, W = (d, w, u) => {
    var s, _;
    const h = N();
    return Array.isArray(h) ? c.value(w) === ((s = h[u]) == null ? void 0 : s.year) && d === ((_ = h[u]) == null ? void 0 : _.month) : !1;
  }, A = (d, w) => {
    if (e.range) {
      const u = N();
      if (Array.isArray(l.value) && Array.isArray(u)) {
        const h = W(d, w, 0) || W(d, w, 1), s = Je(ze(S()), d, c.value(w));
        return Vn(l.value, T.value, s) && !h;
      }
      return !1;
    }
    return !1;
  }, x = H(() => (d) => bt(M.value, (w) => {
    const u = U(d, w.value), h = Ct(
      w.value,
      Ma(c.value(d), e.minDate),
      $a(c.value(d), e.maxDate)
    ) || Yr(e.disabledDates, c.value(d)).includes(w.value), s = A(w.value, d);
    return { active: u, disabled: h, isBetween: s };
  })), Z = H(() => (d) => bt(k.value, (w) => {
    const u = c.value(d) === w.value, h = Ct(w.value, kt(e.minDate), kt(e.maxDate));
    return { active: u, disabled: h };
  })), le = (d, w) => Je(ze(S()), d, c.value(w)), v = (d, w) => {
    const u = l.value ? l.value : ze(/* @__PURE__ */ new Date());
    l.value = Je(u, d, c.value(w)), n("auto-apply");
  }, D = (d, w) => {
    const u = Ca(l, le(d, w), n);
    Un(u, n, e.autoApply, e.modelAuto);
  }, P = (d, w) => {
    Ln(le(d, w), l, e.multiDatesLimit), n("auto-apply", !0);
  };
  return {
    groupedMonths: x,
    groupedYears: Z,
    year: c,
    isDisabled: B,
    defaultedMultiCalendars: a,
    defaultedAriaLabels: t,
    defaultedTransitions: o,
    setHoverDate: (d, w) => {
      T.value = le(d, w);
    },
    selectMonth: (d, w) => (p.value[w].month = d, e.multiDates ? P(d, w) : e.range ? D(d, w) : v(d, w)),
    selectYear: (d, w) => {
      p.value[w].year = d, a.value.count && !a.value.solo && q(w);
    }
  };
}, cl = { class: "dp__month_picker_header" }, fl = ["aria-label", "onClick", "onKeydown"], vl = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "MonthPicker",
  props: {
    ...Xe
  },
  emits: [
    "update:internal-model-value",
    "overlay-closed",
    "reset-flow",
    "range-start",
    "range-end",
    "auto-apply"
  ],
  setup(e, { emit: n }) {
    const a = e, {
      groupedMonths: t,
      groupedYears: o,
      year: l,
      isDisabled: c,
      defaultedMultiCalendars: b,
      defaultedAriaLabels: p,
      defaultedTransitions: M,
      setHoverDate: k,
      selectMonth: T,
      selectYear: Y
    } = dl(a, n), { transitionName: q, showTransition: j } = Yt(M), { showRightIcon: Q, showLeftIcon: B } = qt(), y = z([!1]), N = (A, x) => {
      Y(A, x), W(x);
    }, U = (A, x = !1) => {
      if (!c.value(A, x)) {
        const Z = x ? l.value(A) + 1 : l.value(A) - 1;
        Y(Z, A);
      }
    }, W = (A, x = !1, Z) => {
      x || n("reset-flow"), Z !== void 0 ? y.value[A] = Z : y.value[A] = !y.value[A], y.value || n("overlay-closed");
    };
    return (A, x) => ($(), ve(Hn, {
      "multi-calendars": m(b).count,
      stretch: ""
    }, {
      default: de(({ instance: Z }) => [
        Ke(Rt, {
          items: m(t)(Z),
          "arrow-navigation": A.arrowNavigation,
          "is-last": A.autoApply && !A.keepActionRow,
          "esc-close": A.escClose,
          height: A.modeHeight,
          onSelected: (le) => m(T)(le, Z),
          onHoverValue: (le) => m(k)(le, Z),
          "use-relative": "",
          type: "month"
        }, {
          header: de(() => {
            var le, v, D;
            return [
              ue("div", cl, [
                m(B)(m(b), Z) ? ($(), ve(At, {
                  key: 0,
                  ref: "mpPrevIconRef",
                  "aria-label": (le = m(p)) == null ? void 0 : le.prevYear,
                  disabled: m(c)(Z, !1),
                  onActivate: (P) => U(Z, !1)
                }, {
                  default: de(() => [
                    A.$slots["arrow-left"] ? ee(A.$slots, "arrow-left", { key: 0 }) : I("", !0),
                    A.$slots["arrow-left"] ? I("", !0) : ($(), ve(m(Cn), { key: 1 }))
                  ]),
                  _: 2
                }, 1032, ["aria-label", "disabled", "onActivate"])) : I("", !0),
                ue("div", {
                  class: "dp--year-select",
                  role: "button",
                  ref: "mpYearButtonRef",
                  "aria-label": (v = m(p)) == null ? void 0 : v.openYearsOverlay,
                  tabindex: "0",
                  onClick: () => W(Z, !1),
                  onKeydown: oe(() => W(Z, !1), ["enter"])
                }, [
                  A.$slots.year ? ee(A.$slots, "year", {
                    key: 0,
                    year: m(l)(Z)
                  }) : I("", !0),
                  A.$slots.year ? I("", !0) : ($(), O(ce, { key: 1 }, [
                    nt(Ie(m(l)(Z)), 1)
                  ], 64))
                ], 40, fl),
                m(Q)(m(b), Z) ? ($(), ve(At, {
                  key: 1,
                  ref: "mpNextIconRef",
                  "aria-label": (D = m(p)) == null ? void 0 : D.nextYear,
                  disabled: m(c)(Z, !1),
                  onActivate: (P) => U(Z, !0)
                }, {
                  default: de(() => [
                    A.$slots["arrow-right"] ? ee(A.$slots, "arrow-right", { key: 0 }) : I("", !0),
                    A.$slots["arrow-right"] ? I("", !0) : ($(), ve(m(Rn), { key: 1 }))
                  ]),
                  _: 2
                }, 1032, ["aria-label", "disabled", "onActivate"])) : I("", !0),
                Ke(wt, {
                  name: m(q)(y.value[Z]),
                  css: m(j)
                }, {
                  default: de(() => [
                    y.value[Z] ? ($(), ve(Rt, {
                      key: 0,
                      items: m(o)(Z),
                      "text-input": A.textInput,
                      "esc-close": A.escClose,
                      onToggle: (P) => W(Z),
                      onSelected: (P) => N(P, Z),
                      "is-last": A.autoApply && !A.keepActionRow,
                      type: "year"
                    }, Fe({
                      "button-icon": de(() => [
                        A.$slots["calendar-icon"] ? ee(A.$slots, "calendar-icon", { key: 0 }) : I("", !0),
                        A.$slots["calendar-icon"] ? I("", !0) : ($(), ve(m(It), { key: 1 }))
                      ]),
                      _: 2
                    }, [
                      A.$slots["year-overlay-value"] ? {
                        name: "item",
                        fn: de(({ item: P }) => [
                          ee(A.$slots, "year-overlay-value", {
                            text: P.text,
                            value: P.value
                          })
                        ]),
                        key: "0"
                      } : void 0
                    ]), 1032, ["items", "text-input", "esc-close", "onToggle", "onSelected", "is-last"])) : I("", !0)
                  ]),
                  _: 2
                }, 1032, ["name", "css"])
              ])
            ];
          }),
          _: 2
        }, 1032, ["items", "arrow-navigation", "is-last", "esc-close", "height", "onSelected", "onHoverValue"])
      ]),
      _: 3
    }, 8, ["multi-calendars"]));
  }
}), ml = (e, n) => {
  const { modelValue: a } = Zt(e, n), t = z(null), o = (k) => Array.isArray(a.value) ? a.value.some((T) => fe(T) === k) : a.value ? fe(a.value) === k : !1, l = (k) => e.range && Array.isArray(a.value) ? Vn(a.value, t.value, b(k)) : !1, c = H(() => bt(Fn(e.yearRange, e.reverseYears), (k) => {
    const T = o(k.value), Y = Ct(k.value, kt(e.minDate), kt(e.maxDate)), q = l(k.value);
    return { active: T, disabled: Y, isBetween: q };
  })), b = (k) => dt(ze(/* @__PURE__ */ new Date()), k);
  return {
    groupedYears: c,
    setHoverValue: (k) => {
      t.value = dt(ze(/* @__PURE__ */ new Date()), k);
    },
    selectYear: (k) => {
      if (e.multiDates)
        return Ln(b(k), a, e.multiDatesLimit), n("auto-apply", !0);
      if (e.range) {
        const T = Ca(a, b(k), n);
        return Un(T, n, e.autoApply, e.modelAuto);
      }
      a.value = b(k), n("auto-apply");
    }
  };
}, gl = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "YearPicker",
  props: {
    ...Xe
  },
  emits: ["update:internal-model-value", "reset-flow", "range-start", "range-end", "auto-apply"],
  setup(e, { emit: n }) {
    const a = e, { groupedYears: t, selectYear: o, setHoverValue: l } = ml(a, n);
    return (c, b) => ($(), ve(Rt, {
      items: m(t),
      "is-last": c.autoApply && !c.keepActionRow,
      height: c.modeHeight,
      type: "year",
      "use-relative": "",
      onSelected: m(o),
      onHoverValue: m(l)
    }, Fe({ _: 2 }, [
      c.$slots["year-overlay-value"] ? {
        name: "item",
        fn: de(({ item: p }) => [
          ee(c.$slots, "year-overlay-value", {
            text: p.text,
            value: p.value
          })
        ]),
        key: "0"
      } : void 0
    ]), 1032, ["items", "is-last", "height", "onSelected", "onHoverValue"]));
  }
}), yl = {
  key: 0,
  class: "dp__time_input"
}, hl = ["aria-label", "onKeydown", "onClick"], pl = /* @__PURE__ */ ue("span", { class: "dp__tp_inline_btn_bar dp__tp_btn_in_l" }, null, -1), bl = /* @__PURE__ */ ue("span", { class: "dp__tp_inline_btn_bar dp__tp_btn_in_r" }, null, -1), kl = ["aria-label", "disabled", "onKeydown", "onClick"], wl = ["aria-label", "onKeydown", "onClick"], Dl = /* @__PURE__ */ ue("span", { class: "dp__tp_inline_btn_bar dp__tp_btn_in_l" }, null, -1), Ml = /* @__PURE__ */ ue("span", { class: "dp__tp_inline_btn_bar dp__tp_btn_in_r" }, null, -1), $l = { key: 0 }, Tl = ["aria-label", "onKeydown"], Al = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "TimeInput",
  props: {
    hours: { type: Number, default: 0 },
    minutes: { type: Number, default: 0 },
    seconds: { type: Number, default: 0 },
    closeTimePickerBtn: { type: Object, default: null },
    order: { type: Number, default: 0 },
    disabledTimesConfig: { type: Function, default: null },
    ...Xe
  },
  emits: [
    "set-hours",
    "set-minutes",
    "update:hours",
    "update:minutes",
    "update:seconds",
    "reset-flow",
    "mounted",
    "overlay-closed",
    "am-pm-change"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, { setTimePickerElements: o, setTimePickerBackRef: l } = rt(), { defaultedAriaLabels: c, defaultedTransitions: b, defaultedFilters: p } = Ce(t), { transitionName: M, showTransition: k } = Yt(b), T = Nt({
      hours: !1,
      minutes: !1,
      seconds: !1
    }), Y = z("AM"), q = z(null), j = z([]);
    Ue(() => {
      a("mounted");
    });
    const Q = (r) => De(/* @__PURE__ */ new Date(), {
      hours: r.hours,
      minutes: r.minutes,
      seconds: t.enableSeconds ? r.seconds : 0,
      milliseconds: 0
    }), B = H(() => (r) => D(r, t[r])), y = H(() => ({ hours: t.hours, minutes: t.minutes, seconds: t.seconds })), N = H(() => (r) => !K(+t[r] + +t[`${r}Increment`], r)), U = H(() => (r) => !K(+t[r] - +t[`${r}Increment`], r)), W = (r, L) => ya(De(S(), r), L), A = (r, L) => Ka(De(S(), r), L), x = H(
      () => ({
        dp__time_col: !0,
        dp__time_col_block: !t.timePickerInline,
        dp__time_col_reg_block: !t.enableSeconds && t.is24 && !t.timePickerInline,
        dp__time_col_reg_inline: !t.enableSeconds && t.is24 && t.timePickerInline,
        dp__time_col_reg_with_button: !t.enableSeconds && !t.is24,
        dp__time_col_sec: t.enableSeconds && t.is24,
        dp__time_col_sec_with_button: t.enableSeconds && !t.is24
      })
    ), Z = H(() => {
      const r = [{ type: "hours" }, { type: "", separator: !0 }, { type: "minutes" }];
      return t.enableSeconds ? r.concat([{ type: "", separator: !0 }, { type: "seconds" }]) : r;
    }), le = H(() => Z.value.filter((r) => !r.separator)), v = H(() => (r) => {
      if (r === "hours") {
        const L = u(+t.hours);
        return { text: L < 10 ? `0${L}` : `${L}`, value: L };
      }
      return { text: t[r] < 10 ? `0${t[r]}` : `${t[r]}`, value: t[r] };
    }), D = (r, L) => {
      var g;
      if (!t.disabledTimesConfig)
        return !1;
      const C = t.disabledTimesConfig(t.order, r === "hours" ? L : void 0);
      return C[r] ? !!((g = C[r]) != null && g.includes(L)) : !0;
    }, P = (r) => {
      const L = t.is24 ? 24 : 12, C = r === "hours" ? L : 60, g = +t[`${r}GridIncrement`], F = r === "hours" && !t.is24 ? g : 0, re = [];
      for (let G = F; G < C; G += g)
        re.push({ value: G, text: G < 10 ? `0${G}` : `${G}` });
      return r === "hours" && !t.is24 && re.push({ value: 0, text: "12" }), bt(re, (G) => ({ active: !1, disabled: p.value.times[r].includes(G.value) || !K(G.value, r) || D(r, G.value) }));
    }, K = (r, L) => {
      const C = t.minTime ? Q(cn(t.minTime)) : null, g = t.maxTime ? Q(cn(t.maxTime)) : null, F = Q(cn(y.value, L, r));
      return C && g ? (_t(F, g) || ht(F, g)) && (St(F, C) || ht(F, C)) : C ? St(F, C) || ht(F, C) : g ? _t(F, g) || ht(F, g) : !0;
    }, f = (r) => t[`no${r[0].toUpperCase() + r.slice(1)}Overlay`], E = (r) => {
      f(r) || (T[r] = !T[r], T[r] || a("overlay-closed"));
    }, d = (r) => r === "hours" ? qe : r === "minutes" ? xe : pt, w = (r, L = !0) => {
      const C = L ? W : A, g = L ? +t[`${r}Increment`] : -+t[`${r}Increment`];
      K(+t[r] + g, r) && a(
        `update:${r}`,
        d(r)(C({ [r]: +t[r] }, { [r]: +t[`${r}Increment`] }))
      );
    }, u = (r) => t.is24 ? r : (r >= 12 ? Y.value = "PM" : Y.value = "AM", $r(r)), h = () => {
      Y.value === "PM" ? (Y.value = "AM", a("update:hours", t.hours - 12)) : (Y.value = "PM", a("update:hours", t.hours + 12)), a("am-pm-change", Y.value);
    }, s = (r) => {
      T[r] = !0;
    }, _ = (r, L, C) => {
      if (r && t.arrowNavigation) {
        Array.isArray(j.value[L]) ? j.value[L][C] = r : j.value[L] = [r];
        const g = j.value.reduce(
          (F, re) => re.map((G, ke) => [...F[ke] || [], re[ke]]),
          []
        );
        l(t.closeTimePickerBtn), q.value && (g[1] = g[1].concat(q.value)), o(g, t.order);
      }
    }, te = (r, L) => (E(r), r === "hours" && !t.is24 ? a(`update:${r}`, Y.value === "PM" ? L + 12 : L) : a(`update:${r}`, L));
    return n({ openChildCmp: s }), (r, L) => {
      var C;
      return r.disabled ? I("", !0) : ($(), O("div", yl, [
        ($(!0), O(ce, null, Me(Z.value, (g, F) => {
          var re, G, ke;
          return $(), O("div", {
            key: F,
            class: he(x.value)
          }, [
            g.separator ? ($(), O(ce, { key: 0 }, [
              nt(" : ")
            ], 64)) : ($(), O(ce, { key: 1 }, [
              ue("button", {
                type: "button",
                class: he({
                  dp__btn: !0,
                  dp__inc_dec_button: !t.timePickerInline,
                  dp__inc_dec_button_inline: t.timePickerInline,
                  dp__tp_inline_btn_top: t.timePickerInline,
                  dp__inc_dec_button_disabled: N.value(g.type)
                }),
                "aria-label": (re = m(c)) == null ? void 0 : re.incrementValue(g.type),
                tabindex: "0",
                onKeydown: [
                  oe(pe((se) => w(g.type), ["prevent"]), ["enter"]),
                  oe(pe((se) => w(g.type), ["prevent"]), ["space"])
                ],
                onClick: (se) => w(g.type),
                ref_for: !0,
                ref: (se) => _(se, F, 0)
              }, [
                t.timePickerInline ? ($(), O(ce, { key: 1 }, [
                  pl,
                  bl
                ], 64)) : ($(), O(ce, { key: 0 }, [
                  r.$slots["arrow-up"] ? ee(r.$slots, "arrow-up", { key: 0 }) : I("", !0),
                  r.$slots["arrow-up"] ? I("", !0) : ($(), ve(m(On), { key: 1 }))
                ], 64))
              ], 42, hl),
              ue("button", {
                type: "button",
                "aria-label": (G = m(c)) == null ? void 0 : G.openTpOverlay(g.type),
                class: he({
                  dp__time_display: !0,
                  dp__time_display_block: !t.timePickerInline,
                  dp__time_display_inline: t.timePickerInline,
                  "dp--time-invalid": B.value(g.type),
                  "dp--time-overlay-btn": !B.value(g.type)
                }),
                disabled: f(g.type),
                tabindex: "0",
                onKeydown: [
                  oe(pe((se) => E(g.type), ["prevent"]), ["enter"]),
                  oe(pe((se) => E(g.type), ["prevent"]), ["space"])
                ],
                onClick: (se) => E(g.type),
                ref_for: !0,
                ref: (se) => _(se, F, 1)
              }, [
                r.$slots[g.type] ? ee(r.$slots, g.type, {
                  key: 0,
                  text: v.value(g.type).text,
                  value: v.value(g.type).value
                }) : I("", !0),
                r.$slots[g.type] ? I("", !0) : ($(), O(ce, { key: 1 }, [
                  nt(Ie(v.value(g.type).text), 1)
                ], 64))
              ], 42, kl),
              ue("button", {
                type: "button",
                class: he({
                  dp__btn: !0,
                  dp__inc_dec_button: !t.timePickerInline,
                  dp__inc_dec_button_inline: t.timePickerInline,
                  dp__tp_inline_btn_bottom: t.timePickerInline,
                  dp__inc_dec_button_disabled: U.value(g.type)
                }),
                "aria-label": (ke = m(c)) == null ? void 0 : ke.decrementValue(g.type),
                tabindex: "0",
                onKeydown: [
                  oe(pe((se) => w(g.type, !1), ["prevent"]), ["enter"]),
                  oe(pe((se) => w(g.type, !1), ["prevent"]), ["space"])
                ],
                onClick: (se) => w(g.type, !1),
                ref_for: !0,
                ref: (se) => _(se, F, 2)
              }, [
                t.timePickerInline ? ($(), O(ce, { key: 1 }, [
                  Dl,
                  Ml
                ], 64)) : ($(), O(ce, { key: 0 }, [
                  r.$slots["arrow-down"] ? ee(r.$slots, "arrow-down", { key: 0 }) : I("", !0),
                  r.$slots["arrow-down"] ? I("", !0) : ($(), ve(m(In), { key: 1 }))
                ], 64))
              ], 42, wl)
            ], 64))
          ], 2);
        }), 128)),
        r.is24 ? I("", !0) : ($(), O("div", $l, [
          r.$slots["am-pm-button"] ? ee(r.$slots, "am-pm-button", {
            key: 0,
            toggle: h,
            value: Y.value
          }) : I("", !0),
          r.$slots["am-pm-button"] ? I("", !0) : ($(), O("button", {
            key: 1,
            ref_key: "amPmButton",
            ref: q,
            type: "button",
            class: "dp__pm_am_button",
            role: "button",
            "aria-label": (C = m(c)) == null ? void 0 : C.amPmButton,
            tabindex: "0",
            onClick: h,
            onKeydown: [
              oe(pe(h, ["prevent"]), ["enter"]),
              oe(pe(h, ["prevent"]), ["space"])
            ]
          }, Ie(Y.value), 41, Tl))
        ])),
        ($(!0), O(ce, null, Me(le.value, (g, F) => ($(), ve(wt, {
          key: F,
          name: m(M)(T[g.type]),
          css: m(k)
        }, {
          default: de(() => [
            T[g.type] ? ($(), ve(Rt, {
              key: 0,
              items: P(g.type),
              "is-last": r.autoApply && !r.keepActionRow,
              "esc-close": r.escClose,
              type: g.type,
              "text-input": r.textInput,
              "arrow-navigation": r.arrowNavigation,
              onSelected: (re) => te(g.type, re),
              onToggle: (re) => E(g.type),
              onResetFlow: L[0] || (L[0] = (re) => r.$emit("reset-flow"))
            }, Fe({
              "button-icon": de(() => [
                r.$slots["clock-icon"] ? ee(r.$slots, "clock-icon", { key: 0 }) : I("", !0),
                r.$slots["clock-icon"] ? I("", !0) : ($(), ve(m(Nn), { key: 1 }))
              ]),
              _: 2
            }, [
              r.$slots[`${g.type}-overlay-value`] ? {
                name: "item",
                fn: de(({ item: re }) => [
                  ee(r.$slots, `${g.type}-overlay-value`, {
                    text: re.text,
                    value: re.value
                  })
                ]),
                key: "0"
              } : void 0
            ]), 1032, ["items", "is-last", "esc-close", "type", "text-input", "arrow-navigation", "onSelected", "onToggle"])) : I("", !0)
          ]),
          _: 2
        }, 1032, ["name", "css"]))), 128))
      ]));
    };
  }
}), _l = ["aria-label"], Sl = ["tabindex"], Pl = ["aria-label"], Ra = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "TimePicker",
  props: {
    hours: { type: [Number, Array], default: 0 },
    minutes: { type: [Number, Array], default: 0 },
    seconds: { type: [Number, Array], default: 0 },
    disabledTimesConfig: { type: Function, default: null },
    ...Xe
  },
  emits: [
    "update:hours",
    "update:minutes",
    "update:seconds",
    "mount",
    "reset-flow",
    "overlay-opened",
    "overlay-closed",
    "am-pm-change"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, { buildMatrix: o, setTimePicker: l } = rt(), c = Ot(), { defaultedTransitions: b, defaultedAriaLabels: p, defaultedTextInput: M } = Ce(t), { transitionName: k, showTransition: T } = Yt(b), { hideNavigationButtons: Y } = qt(), q = z(null), j = z(null), Q = z([]), B = z(null);
    Ue(() => {
      a("mount"), !t.timePicker && t.arrowNavigation ? o([Ae(q.value)], "time") : l(!0, t.timePicker);
    });
    const y = H(() => t.range && t.modelAuto ? Da(t.internalModelValue) : !0), N = z(!1), U = (f) => ({
      hours: Array.isArray(t.hours) ? t.hours[f] : t.hours,
      minutes: Array.isArray(t.minutes) ? t.minutes[f] : t.minutes,
      seconds: Array.isArray(t.seconds) ? t.seconds[f] : t.seconds
    }), W = H(() => {
      const f = [];
      if (t.range)
        for (let E = 0; E < 2; E++)
          f.push(U(E));
      else
        f.push(U(0));
      return f;
    }), A = (f, E = !1, d = "") => {
      E || a("reset-flow"), N.value = f, a(f ? "overlay-opened" : "overlay-closed"), t.arrowNavigation && l(f), at(() => {
        d !== "" && Q.value[0] && Q.value[0].openChildCmp(d);
      });
    }, x = H(() => ({
      dp__btn: !0,
      dp__button: !0,
      dp__button_bottom: t.autoApply && !t.keepActionRow
    })), Z = je(c, "timePicker"), le = (f, E, d) => t.range ? E === 0 ? [f, W.value[1][d]] : [W.value[0][d], f] : f, v = (f) => {
      a("update:hours", f);
    }, D = (f) => {
      a("update:minutes", f);
    }, P = (f) => {
      a("update:seconds", f);
    }, K = () => {
      if (B.value && !M.value.enabled) {
        const f = Ar(B.value);
        f && f.focus({ preventScroll: !0 });
      }
    };
    return n({ toggleTimePicker: A }), (f, E) => {
      var d;
      return $(), O("div", null, [
        !f.timePicker && !f.timePickerInline ? Ut(($(), O("button", {
          key: 0,
          type: "button",
          class: he(x.value),
          "aria-label": (d = m(p)) == null ? void 0 : d.openTimePicker,
          tabindex: "0",
          ref_key: "openTimePickerBtn",
          ref: q,
          onKeydown: [
            E[0] || (E[0] = oe((w) => A(!0), ["enter"])),
            E[1] || (E[1] = oe((w) => A(!0), ["space"]))
          ],
          onClick: E[2] || (E[2] = (w) => A(!0))
        }, [
          f.$slots["clock-icon"] ? ee(f.$slots, "clock-icon", { key: 0 }) : I("", !0),
          f.$slots["clock-icon"] ? I("", !0) : ($(), ve(m(Nn), { key: 1 }))
        ], 42, _l)), [
          [Wt, !m(Y)(f.hideNavigation, "time")]
        ]) : I("", !0),
        Ke(wt, {
          name: m(k)(N.value),
          css: m(T) && !f.timePickerInline
        }, {
          default: de(() => {
            var w;
            return [
              N.value || f.timePicker || f.timePickerInline ? ($(), O("div", {
                key: 0,
                class: he({
                  dp__overlay: !f.timePickerInline,
                  "dp--overlay-absolute": !t.timePicker && !f.timePickerInline,
                  "dp--overlay-relative": t.timePicker
                }),
                style: Ge(f.timePicker ? { height: `${f.modeHeight}px` } : void 0),
                ref_key: "overlayRef",
                ref: B,
                tabindex: f.timePickerInline ? void 0 : 0
              }, [
                ue("div", {
                  class: he(
                    f.timePickerInline ? "dp__time_picker_inline_container" : "dp__overlay_container dp__container_flex dp__time_picker_overlay_container"
                  ),
                  style: { display: "flex" }
                }, [
                  f.$slots["time-picker-overlay"] ? ee(f.$slots, "time-picker-overlay", {
                    key: 0,
                    hours: e.hours,
                    minutes: e.minutes,
                    seconds: e.seconds,
                    setHours: v,
                    setMinutes: D,
                    setSeconds: P
                  }) : I("", !0),
                  f.$slots["time-picker-overlay"] ? I("", !0) : ($(), O("div", {
                    key: 1,
                    class: he(f.timePickerInline ? "dp__flex" : "dp__overlay_row dp__flex_row")
                  }, [
                    ($(!0), O(ce, null, Me(W.value, (u, h) => Ut(($(), ve(Al, Ee({ key: h }, {
                      ...f.$props,
                      order: h,
                      hours: u.hours,
                      minutes: u.minutes,
                      seconds: u.seconds,
                      closeTimePickerBtn: j.value,
                      disabledTimesConfig: e.disabledTimesConfig,
                      disabled: h === 0 ? f.fixedStart : f.fixedEnd
                    }, {
                      ref_for: !0,
                      ref_key: "timeInputRefs",
                      ref: Q,
                      "onUpdate:hours": (s) => v(le(s, h, "hours")),
                      "onUpdate:minutes": (s) => D(le(s, h, "minutes")),
                      "onUpdate:seconds": (s) => P(le(s, h, "seconds")),
                      onMounted: K,
                      onOverlayClosed: K,
                      onAmPmChange: E[3] || (E[3] = (s) => f.$emit("am-pm-change", s))
                    }), Fe({ _: 2 }, [
                      Me(m(Z), (s, _) => ({
                        name: s,
                        fn: de((te) => [
                          ee(f.$slots, s, Re(He(te)))
                        ])
                      }))
                    ]), 1040, ["onUpdate:hours", "onUpdate:minutes", "onUpdate:seconds"])), [
                      [Wt, h === 0 ? !0 : y.value]
                    ])), 128))
                  ], 2)),
                  !f.timePicker && !f.timePickerInline ? Ut(($(), O("button", {
                    key: 2,
                    type: "button",
                    ref_key: "closeTimePickerBtn",
                    ref: j,
                    class: he(x.value),
                    "aria-label": (w = m(p)) == null ? void 0 : w.closeTimePicker,
                    tabindex: "0",
                    onKeydown: [
                      E[4] || (E[4] = oe((u) => A(!1), ["enter"])),
                      E[5] || (E[5] = oe((u) => A(!1), ["space"]))
                    ],
                    onClick: E[6] || (E[6] = (u) => A(!1))
                  }, [
                    f.$slots["calendar-icon"] ? ee(f.$slots, "calendar-icon", { key: 0 }) : I("", !0),
                    f.$slots["calendar-icon"] ? I("", !0) : ($(), ve(m(It), { key: 1 }))
                  ], 42, Pl)), [
                    [Wt, !m(Y)(f.hideNavigation, "time")]
                  ]) : I("", !0)
                ], 2)
              ], 14, Sl)) : I("", !0)
            ];
          }),
          _: 3
        }, 8, ["name", "css"])
      ]);
    };
  }
}), Na = (e, n, a, t) => {
  const o = (y, N) => Array.isArray(n[y]) ? n[y][N] : n[y], l = (y) => e.enableSeconds ? Array.isArray(n.seconds) ? n.seconds[y] : n.seconds : 0, c = (y, N) => y ? N !== void 0 ? tt(y, o("hours", N), o("minutes", N), l(N)) : tt(y, n.hours, n.minutes, l()) : S(), b = (y, N) => {
    n[y] = N;
  }, p = (y, N) => {
    const U = Object.fromEntries(
      Object.keys(n).map((W) => W === y ? [W, N] : [W, n[W]].slice())
    );
    if (e.range && !e.disableTimeRangeValidation) {
      const W = (x) => a.value ? tt(
        a.value[x],
        U.hours[x],
        U.minutes[x],
        U.seconds[x]
      ) : null, A = (x) => Pn(a.value[x], 0);
      return !(ye(W(0), W(1)) && (St(W(0), A(1)) || _t(W(1), A(0))));
    }
    return !0;
  }, M = (y, N) => {
    p(y, N) && (b(y, N), t && t());
  }, k = (y) => {
    M("hours", y);
  }, T = (y) => {
    M("minutes", y);
  }, Y = (y) => {
    M("seconds", y);
  }, q = (y, N, U, W) => {
    N && k(y), !N && !U && T(y), U && Y(y), a.value && W(a.value);
  }, j = (y) => {
    if (y) {
      const N = Array.isArray(y), U = N ? [+y[0].hours, +y[1].hours] : +y.hours, W = N ? [+y[0].minutes, +y[1].minutes] : +y.minutes, A = N ? [+y[0].seconds, +y[1].seconds] : +y.seconds;
      b("hours", U), b("minutes", W), e.enableSeconds && b("seconds", A);
    }
  }, Q = (y, N) => {
    const U = {
      hours: Array.isArray(n.hours) ? n.hours[y] : n.hours,
      disabledArr: []
    };
    return (N || N === 0) && (U.hours = N), Array.isArray(e.disabledTimes) && (U.disabledArr = e.range && Array.isArray(e.disabledTimes[y]) ? e.disabledTimes[y] : e.disabledTimes), U;
  }, B = H(() => (y, N) => {
    var U;
    if (Array.isArray(e.disabledTimes)) {
      const { disabledArr: W, hours: A } = Q(y, N), x = W.filter((Z) => +Z.hours === A);
      return ((U = x[0]) == null ? void 0 : U.minutes) === "*" ? { hours: [A], minutes: void 0, seconds: void 0 } : {
        hours: [],
        minutes: (x == null ? void 0 : x.map((Z) => +Z.minutes)) ?? [],
        seconds: (x == null ? void 0 : x.map((Z) => Z.seconds ? +Z.seconds : void 0)) ?? []
      };
    }
    return { hours: [], minutes: [], seconds: [] };
  });
  return {
    setTime: b,
    updateHours: k,
    updateMinutes: T,
    updateSeconds: Y,
    getSetDateTime: c,
    updateTimeValues: q,
    getSecondsValue: l,
    assignStartTime: j,
    disabledTimesConfig: B
  };
}, Cl = (e, n) => {
  const { modelValue: a, time: t } = Zt(e, n), { defaultedStartTime: o } = Ce(e), { updateTimeValues: l, getSetDateTime: c, setTime: b, assignStartTime: p, disabledTimesConfig: M } = Na(
    e,
    t,
    a
  ), k = () => {
    e.range ? a.value = [c(null, 0), c(null, 1)] : a.value = c(null);
  }, T = (B) => Array.isArray(B) ? [ft(S(B[0])), ft(S(B[1]))] : [ft(B ?? S())], Y = (B, y, N) => {
    b("hours", B), b("minutes", y), e.enableSeconds && b("seconds", N);
  }, q = () => {
    const [B, y] = T(a.value);
    return e.range ? Y(
      [B.hours, y.hours],
      [B.minutes, y.minutes],
      [B.seconds, y.minutes]
    ) : Y(B.hours, B.minutes, B.seconds);
  };
  Ue(() => {
    if (!e.shadow)
      return p(o.value), a.value ? q() : k();
  });
  const j = () => {
    Array.isArray(a.value) ? a.value = a.value.map((B, y) => B && c(B, y)) : a.value = c(a.value), n("time-update");
  };
  return {
    time: t,
    disabledTimesConfig: M,
    updateTime: (B, y = !0, N = !1) => {
      l(B, y, N, j);
    }
  };
}, Rl = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "TimePickerSolo",
  props: {
    ...Xe
  },
  emits: ["update:internal-model-value", "time-update", "am-pm-change"],
  setup(e, { emit: n }) {
    const a = e, t = Ot(), o = je(t, "timePicker"), { time: l, disabledTimesConfig: c, updateTime: b } = Cl(a, n);
    return (p, M) => ($(), ve(Hn, {
      "multi-calendars": 0,
      stretch: ""
    }, {
      default: de(() => [
        Ke(Ra, Ee(p.$props, {
          hours: m(l).hours,
          minutes: m(l).minutes,
          seconds: m(l).seconds,
          "internal-model-value": p.internalModelValue,
          "disabled-times-config": m(c),
          "onUpdate:hours": M[0] || (M[0] = (k) => m(b)(k)),
          "onUpdate:minutes": M[1] || (M[1] = (k) => m(b)(k, !1)),
          "onUpdate:seconds": M[2] || (M[2] = (k) => m(b)(k, !1, !0)),
          onAmPmChange: M[3] || (M[3] = (k) => p.$emit("am-pm-change", k))
        }), Fe({ _: 2 }, [
          Me(m(o), (k, T) => ({
            name: k,
            fn: de((Y) => [
              ee(p.$slots, k, Re(He(Y)))
            ])
          }))
        ]), 1040, ["hours", "minutes", "seconds", "internal-model-value", "disabled-times-config"])
      ]),
      _: 3
    }));
  }
}), Nl = { class: "dp__month_year_row" }, Ol = ["onClick", "onKeydown", "aria-label"], Il = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "DpHeader",
  props: {
    month: { type: Number, default: 0 },
    year: { type: Number, default: 0 },
    instance: { type: Number, default: 0 },
    years: { type: Array, default: () => [] },
    months: { type: Array, default: () => [] },
    ...Xe
  },
  emits: ["update-month-year", "mount", "reset-flow", "overlay-closed"],
  setup(e, { expose: n, emit: a }) {
    const t = e, { defaultedTransitions: o, defaultedAriaLabels: l, defaultedMultiCalendars: c, defaultedFilters: b } = Ce(t), { transitionName: p, showTransition: M } = Yt(o), { buildMatrix: k } = rt(), { handleMonthYearChange: T, isDisabled: Y, updateMonthYear: q } = jr(t, a), { showLeftIcon: j, showRightIcon: Q } = qt(), B = z(!1), y = z(!1), N = z([null, null, null, null]);
    Ue(() => {
      a("mount");
    });
    const U = (u) => ({
      get: () => t[u],
      set: (h) => {
        const s = u === We.month ? We.year : We.month;
        a("update-month-year", { [u]: h, [s]: t[s] }), u === We.month ? P(!0) : K(!0);
      }
    }), W = H(U(We.month)), A = H(U(We.year)), x = H(() => (u) => ({
      month: t.month,
      year: t.year,
      items: u === We.month ? t.months : t.years,
      instance: t.instance,
      updateMonthYear: q,
      toggle: u === We.month ? P : K
    })), Z = H(() => {
      const u = t.months.find((h) => h.value === t.month);
      return u || { text: "", value: 0 };
    }), le = H(() => bt(t.months, (u) => {
      const h = t.month === u.value, s = Ct(
        u.value,
        Ma(t.year, t.minDate),
        $a(t.year, t.maxDate)
      ) || b.value.months.includes(u.value);
      return { active: h, disabled: s };
    })), v = H(() => bt(t.years, (u) => {
      const h = t.year === u.value, s = Ct(u.value, kt(t.minDate), kt(t.maxDate)) || b.value.years.includes(u.value);
      return { active: h, disabled: s };
    })), D = (u, h) => {
      h !== void 0 ? u.value = h : u.value = !u.value, u.value || a("overlay-closed");
    }, P = (u = !1, h) => {
      f(u), D(B, h);
    }, K = (u = !1, h) => {
      f(u), D(y, h);
    }, f = (u) => {
      u || a("reset-flow");
    }, E = (u, h) => {
      t.arrowNavigation && (N.value[h] = Ae(u), k(N.value, "monthYear"));
    }, d = H(() => {
      var u, h;
      return [
        {
          type: We.month,
          index: 1,
          toggle: P,
          modelValue: W.value,
          updateModelValue: (s) => W.value = s,
          text: Z.value.text,
          showSelectionGrid: B.value,
          items: le.value,
          ariaLabel: (u = l.value) == null ? void 0 : u.openMonthsOverlay
        },
        {
          type: We.year,
          index: 2,
          toggle: K,
          modelValue: A.value,
          updateModelValue: (s) => A.value = s,
          text: t.year,
          showSelectionGrid: y.value,
          items: v.value,
          ariaLabel: (h = l.value) == null ? void 0 : h.openYearsOverlay
        }
      ];
    }), w = H(
      () => t.disableYearSelect ? [d.value[0]] : d.value
    );
    return n({
      toggleMonthPicker: P,
      toggleYearPicker: K,
      handleMonthYearChange: T
    }), (u, h) => {
      var s, _, te;
      return $(), O("div", Nl, [
        u.$slots["month-year"] ? ee(u.$slots, "month-year", Re(Ee({ key: 0 }, { month: e.month, year: e.year, months: e.months, years: e.years, updateMonthYear: m(q), handleMonthYearChange: m(T), instance: e.instance }))) : ($(), O(ce, { key: 1 }, [
          m(j)(m(c), e.instance) && !u.vertical ? ($(), ve(At, {
            key: 0,
            "aria-label": (s = m(l)) == null ? void 0 : s.prevMonth,
            disabled: m(Y)(!1),
            onActivate: h[0] || (h[0] = (r) => m(T)(!1)),
            onSetRef: h[1] || (h[1] = (r) => E(r, 0))
          }, {
            default: de(() => [
              u.$slots["arrow-left"] ? ee(u.$slots, "arrow-left", { key: 0 }) : I("", !0),
              u.$slots["arrow-left"] ? I("", !0) : ($(), ve(m(Cn), { key: 1 }))
            ]),
            _: 3
          }, 8, ["aria-label", "disabled"])) : I("", !0),
          ue("div", {
            class: he(["dp__month_year_wrap", {
              dp__year_disable_select: u.disableYearSelect
            }])
          }, [
            ($(!0), O(ce, null, Me(w.value, (r, L) => ($(), O(ce, {
              key: r.type
            }, [
              ue("button", {
                type: "button",
                class: "dp__btn dp__month_year_select",
                onClick: r.toggle,
                onKeydown: [
                  oe(pe(r.toggle, ["prevent"]), ["enter"]),
                  oe(pe(r.toggle, ["prevent"]), ["space"])
                ],
                "aria-label": r.ariaLabel,
                tabindex: "0",
                ref_for: !0,
                ref: (C) => E(C, L + 1)
              }, [
                u.$slots[r.type] ? ee(u.$slots, r.type, Re(Ee({ key: 0 }, { text: r.text }))) : I("", !0),
                u.$slots[r.type] ? I("", !0) : ($(), O(ce, { key: 1 }, [
                  nt(Ie(r.text), 1)
                ], 64))
              ], 40, Ol),
              Ke(wt, {
                name: m(p)(r.showSelectionGrid),
                css: m(M)
              }, {
                default: de(() => [
                  r.showSelectionGrid ? ($(), ve(Rt, {
                    key: 0,
                    items: r.items,
                    "arrow-navigation": u.arrowNavigation,
                    "hide-navigation": u.hideNavigation,
                    "is-last": u.autoApply && !u.keepActionRow,
                    "skip-button-ref": !1,
                    type: r.type,
                    "header-refs": [],
                    "esc-close": u.escClose,
                    "text-input": u.textInput,
                    onSelected: r.updateModelValue,
                    onToggle: r.toggle
                  }, Fe({
                    "button-icon": de(() => [
                      u.$slots["calendar-icon"] ? ee(u.$slots, "calendar-icon", { key: 0 }) : I("", !0),
                      u.$slots["calendar-icon"] ? I("", !0) : ($(), ve(m(It), { key: 1 }))
                    ]),
                    _: 2
                  }, [
                    u.$slots[`${r.type}-overlay-val`] ? {
                      name: "item",
                      fn: de(({ item: C }) => [
                        ee(u.$slots, `${r.type}-overlay-val`, {
                          text: C.text,
                          value: C.value
                        })
                      ]),
                      key: "0"
                    } : void 0,
                    u.$slots[`${r.type}-overlay`] ? {
                      name: "overlay",
                      fn: de(() => [
                        ee(u.$slots, `${r.type}-overlay`, Re(He(x.value(r.type))))
                      ]),
                      key: "1"
                    } : void 0,
                    u.$slots[`${r.type}-overlay-header`] ? {
                      name: "header",
                      fn: de(() => [
                        ee(u.$slots, `${r.type}-overlay-header`, {
                          toggle: r.toggle
                        })
                      ]),
                      key: "2"
                    } : void 0
                  ]), 1032, ["items", "arrow-navigation", "hide-navigation", "is-last", "type", "esc-close", "text-input", "onSelected", "onToggle"])) : I("", !0)
                ]),
                _: 2
              }, 1032, ["name", "css"])
            ], 64))), 128))
          ], 2),
          m(j)(m(c), e.instance) && u.vertical ? ($(), ve(At, {
            key: 1,
            "aria-label": (_ = m(l)) == null ? void 0 : _.prevMonth,
            disabled: m(Y)(!1),
            onActivate: h[2] || (h[2] = (r) => m(T)(!1))
          }, {
            default: de(() => [
              u.$slots["arrow-up"] ? ee(u.$slots, "arrow-up", { key: 0 }) : I("", !0),
              u.$slots["arrow-up"] ? I("", !0) : ($(), ve(m(On), { key: 1 }))
            ]),
            _: 3
          }, 8, ["aria-label", "disabled"])) : I("", !0),
          m(Q)(m(c), e.instance) ? ($(), ve(At, {
            key: 2,
            ref: "rightIcon",
            disabled: m(Y)(!0),
            "aria-label": (te = m(l)) == null ? void 0 : te.nextMonth,
            onActivate: h[3] || (h[3] = (r) => m(T)(!0)),
            onSetRef: h[4] || (h[4] = (r) => E(r, u.disableYearSelect ? 2 : 3))
          }, {
            default: de(() => [
              u.$slots[u.vertical ? "arrow-down" : "arrow-right"] ? ee(u.$slots, u.vertical ? "arrow-down" : "arrow-right", { key: 0 }) : I("", !0),
              u.$slots[u.vertical ? "arrow-down" : "arrow-right"] ? I("", !0) : ($(), ve(Sn(u.vertical ? m(In) : m(Rn)), { key: 1 }))
            ]),
            _: 3
          }, 8, ["disabled", "aria-label"])) : I("", !0)
        ], 64))
      ]);
    };
  }
}), Yl = ["aria-label"], Bl = {
  class: "dp__calendar_header",
  role: "row"
}, El = {
  key: 0,
  class: "dp__calendar_header_item",
  role: "gridcell"
}, Fl = /* @__PURE__ */ ue("div", { class: "dp__calendar_header_separator" }, null, -1), Vl = ["aria-label"], Hl = {
  key: 0,
  role: "gridcell",
  class: "dp__calendar_item dp__week_num"
}, Ll = { class: "dp__cell_inner" }, Ul = ["aria-selected", "aria-disabled", "aria-label", "onClick", "onKeydown", "onMouseenter", "onMouseleave"], Wl = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "DpCalendar",
  props: {
    mappedDates: { type: Array, default: () => [] },
    instance: { type: Number, default: 0 },
    month: { type: Number, default: 0 },
    year: { type: Number, default: 0 },
    ...Xe
  },
  emits: [
    "select-date",
    "set-hover-date",
    "handle-scroll",
    "mount",
    "handle-swipe",
    "handle-space",
    "tooltip-open",
    "tooltip-close"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, { buildMultiLevelMatrix: o } = rt(), { defaultedTransitions: l, defaultedAriaLabels: c, defaultedMultiCalendars: b } = Ce(t), p = z(null), M = z({
      bottom: "",
      left: "",
      transform: ""
    }), k = z([]), T = z(null), Y = z(!0), q = z(""), j = z({ startX: 0, endX: 0, startY: 0, endY: 0 }), Q = z([]), B = z({ left: "50%" }), y = H(() => t.calendar ? t.calendar(t.mappedDates) : t.mappedDates), N = H(() => t.dayNames ? Array.isArray(t.dayNames) ? t.dayNames : t.dayNames(t.locale, +t.weekStart) : Mr(t.formatLocale, t.locale, +t.weekStart));
    Ue(() => {
      a("mount", { cmp: "calendar", refs: k }), t.noSwipe || T.value && (T.value.addEventListener("touchstart", K, { passive: !1 }), T.value.addEventListener("touchend", f, { passive: !1 }), T.value.addEventListener("touchmove", E, { passive: !1 })), t.monthChangeOnScroll && T.value && T.value.addEventListener("wheel", u, { passive: !1 });
    });
    const U = (s) => s ? t.vertical ? "vNext" : "next" : t.vertical ? "vPrevious" : "previous", W = (s, _) => {
      if (t.transitions) {
        const te = Le(Je(S(), t.month, t.year));
        q.value = Ne(Le(Je(S(), s, _)), te) ? l.value[U(!0)] : l.value[U(!1)], Y.value = !1, at(() => {
          Y.value = !0;
        });
      }
    }, A = H(
      () => ({
        [t.calendarClassName]: !!t.calendarClassName
      })
    ), x = H(() => (s) => {
      const _ = Tr(s);
      return {
        dp__marker_dot: _.type === "dot",
        dp__marker_line: _.type === "line"
      };
    }), Z = H(() => (s) => ye(s, p.value)), le = H(() => ({
      dp__calendar: !0,
      dp__calendar_next: b.value.count > 0 && t.instance !== 0
    })), v = H(() => (s) => t.hideOffsetDates ? s.current : !0), D = async (s, _, te) => {
      var r, L;
      if (a("set-hover-date", s), (L = (r = s.marker) == null ? void 0 : r.tooltip) != null && L.length) {
        const C = Ae(k.value[_][te]);
        if (C) {
          const { width: g, height: F } = C.getBoundingClientRect();
          p.value = s.value;
          let re = { left: `${g / 2}px` }, G = -50;
          if (await at(), Q.value[0]) {
            const { left: ke, width: se } = Q.value[0].getBoundingClientRect();
            ke < 0 && (re = { left: "0" }, G = 0, B.value.left = `${g / 2}px`), window.innerWidth < ke + se && (re = { right: "0" }, G = 0, B.value.left = `${se - g / 2}px`);
          }
          M.value = {
            bottom: `${F}px`,
            ...re,
            transform: `translateX(${G}%)`
          }, a("tooltip-open", s.marker);
        }
      }
    }, P = (s) => {
      p.value && (p.value = null, M.value = JSON.parse(JSON.stringify({ bottom: "", left: "", transform: "" })), a("tooltip-close", s.marker));
    }, K = (s) => {
      j.value.startX = s.changedTouches[0].screenX, j.value.startY = s.changedTouches[0].screenY;
    }, f = (s) => {
      j.value.endX = s.changedTouches[0].screenX, j.value.endY = s.changedTouches[0].screenY, d();
    }, E = (s) => {
      t.vertical && !t.inline && s.preventDefault();
    }, d = () => {
      const s = t.vertical ? "Y" : "X";
      Math.abs(j.value[`start${s}`] - j.value[`end${s}`]) > 10 && a("handle-swipe", j.value[`start${s}`] > j.value[`end${s}`] ? "right" : "left");
    }, w = (s, _, te) => {
      s && (Array.isArray(k.value[_]) ? k.value[_][te] = s : k.value[_] = [s]), t.arrowNavigation && o(k.value, "calendar");
    }, u = (s) => {
      t.monthChangeOnScroll && (s.preventDefault(), a("handle-scroll", s));
    }, h = (s) => {
      const _ = s[0];
      return t.weekNumbers === "local" ? Ga(_.value, { weekStartsOn: +t.weekStart }) : t.weekNumbers === "iso" ? Za(_.value) : typeof t.weekNumbers == "function" ? t.weekNumbers(_.value) : "";
    };
    return n({ triggerTransition: W }), (s, _) => {
      var te;
      return $(), O("div", {
        class: he(le.value)
      }, [
        ue("div", {
          ref_key: "calendarWrapRef",
          ref: T,
          role: "grid",
          class: he(A.value),
          "aria-label": (te = m(c)) == null ? void 0 : te.calendarWrap
        }, [
          ($(), O(ce, { key: 0 }, [
            ue("div", Bl, [
              s.weekNumbers ? ($(), O("div", El, Ie(s.weekNumName), 1)) : I("", !0),
              ($(!0), O(ce, null, Me(N.value, (r, L) => ($(), O("div", {
                class: "dp__calendar_header_item",
                role: "gridcell",
                key: L
              }, [
                s.$slots["calendar-header"] ? ee(s.$slots, "calendar-header", {
                  key: 0,
                  day: r,
                  index: L
                }) : I("", !0),
                s.$slots["calendar-header"] ? I("", !0) : ($(), O(ce, { key: 1 }, [
                  nt(Ie(r), 1)
                ], 64))
              ]))), 128))
            ]),
            Fl,
            Ke(wt, {
              name: q.value,
              css: !!s.transitions
            }, {
              default: de(() => {
                var r;
                return [
                  Y.value ? ($(), O("div", {
                    key: 0,
                    class: "dp__calendar",
                    role: "grid",
                    "aria-label": ((r = m(c)) == null ? void 0 : r.calendarDays) || void 0
                  }, [
                    ($(!0), O(ce, null, Me(y.value, (L, C) => ($(), O("div", {
                      class: "dp__calendar_row",
                      role: "row",
                      key: C
                    }, [
                      s.weekNumbers ? ($(), O("div", Hl, [
                        ue("div", Ll, Ie(h(L.days)), 1)
                      ])) : I("", !0),
                      ($(!0), O(ce, null, Me(L.days, (g, F) => {
                        var re, G, ke;
                        return $(), O("div", {
                          role: "gridcell",
                          class: "dp__calendar_item",
                          ref_for: !0,
                          ref: (se) => w(se, C, F),
                          key: F + C,
                          "aria-selected": g.classData.dp__active_date || g.classData.dp__range_start || g.classData.dp__range_start,
                          "aria-disabled": g.classData.dp__cell_disabled || void 0,
                          "aria-label": (G = (re = m(c)) == null ? void 0 : re.day) == null ? void 0 : G.call(re, g),
                          tabindex: "0",
                          onClick: pe((se) => s.$emit("select-date", g), ["stop", "prevent"]),
                          onKeydown: [
                            oe((se) => s.$emit("select-date", g), ["enter"]),
                            oe((se) => s.$emit("handle-space", g), ["space"])
                          ],
                          onMouseenter: (se) => D(g, C, F),
                          onMouseleave: (se) => P(g)
                        }, [
                          ue("div", {
                            class: he(["dp__cell_inner", g.classData])
                          }, [
                            s.$slots.day && v.value(g) ? ee(s.$slots, "day", {
                              key: 0,
                              day: +g.text,
                              date: g.value
                            }) : I("", !0),
                            s.$slots.day ? I("", !0) : ($(), O(ce, { key: 1 }, [
                              nt(Ie(g.text), 1)
                            ], 64)),
                            g.marker && v.value(g) ? ($(), O(ce, { key: 2 }, [
                              s.$slots.marker ? ee(s.$slots, "marker", {
                                key: 0,
                                marker: g.marker,
                                day: +g.text,
                                date: g.value
                              }) : ($(), O("div", {
                                key: 1,
                                class: he(x.value(g.marker)),
                                style: Ge(g.marker.color ? { backgroundColor: g.marker.color } : {})
                              }, null, 6))
                            ], 64)) : I("", !0),
                            Z.value(g.value) ? ($(), O("div", {
                              key: 3,
                              class: "dp__marker_tooltip",
                              ref_for: !0,
                              ref_key: "activeTooltip",
                              ref: Q,
                              style: Ge(M.value)
                            }, [
                              (ke = g.marker) != null && ke.tooltip ? ($(), O("div", {
                                key: 0,
                                class: "dp__tooltip_content",
                                onClick: _[0] || (_[0] = pe(() => {
                                }, ["stop"]))
                              }, [
                                ($(!0), O(ce, null, Me(g.marker.tooltip, (se, R) => ($(), O("div", {
                                  key: R,
                                  class: "dp__tooltip_text"
                                }, [
                                  s.$slots["marker-tooltip"] ? ee(s.$slots, "marker-tooltip", {
                                    key: 0,
                                    tooltip: se,
                                    day: g.value
                                  }) : I("", !0),
                                  s.$slots["marker-tooltip"] ? I("", !0) : ($(), O(ce, { key: 1 }, [
                                    ue("div", {
                                      class: "dp__tooltip_mark",
                                      style: Ge(se.color ? { backgroundColor: se.color } : {})
                                    }, null, 4),
                                    ue("div", null, Ie(se.text), 1)
                                  ], 64))
                                ]))), 128)),
                                ue("div", {
                                  class: "dp__arrow_bottom_tp",
                                  style: Ge(B.value)
                                }, null, 4)
                              ])) : I("", !0)
                            ], 4)) : I("", !0)
                          ], 2)
                        ], 40, Ul);
                      }), 128))
                    ]))), 128))
                  ], 8, Vl)) : I("", !0)
                ];
              }),
              _: 3
            }, 8, ["name", "css"])
          ], 64))
        ], 10, Yl)
      ], 2);
    };
  }
}), sa = (e) => Array.isArray(e), zl = (e, n, a, t) => {
  const o = z([]), { modelValue: l, calendars: c, time: b } = Zt(e, n), { defaultedMultiCalendars: p, defaultedStartTime: M } = Ce(e), { validateMonthYearInRange: k, isDisabled: T, isDateRangeAllowed: Y, checkMinMaxRange: q } = Bt(e), { updateTimeValues: j, getSetDateTime: Q, setTime: B, assignStartTime: y, disabledTimesConfig: N } = Na(
    e,
    b,
    l,
    t
  ), U = H(
    () => (i) => c.value[i] ? c.value[i].month : 0
  ), W = H(
    () => (i) => c.value[i] ? c.value[i].year : 0
  ), A = (i, V, ne) => {
    var me, we;
    c.value[i] || (c.value[i] = { month: 0, year: 0 }), c.value[i].month = ta(V) ? (me = c.value[i]) == null ? void 0 : me.month : V, c.value[i].year = ta(ne) ? (we = c.value[i]) == null ? void 0 : we.year : ne;
  }, x = () => {
    e.autoApply && n("select-date");
  };
  mt(l, (i, V) => {
    JSON.stringify(i) !== JSON.stringify(V) && v();
  }), Ue(() => {
    e.shadow || (l.value || (s(), M.value && y(M.value)), v(!0), e.focusStartDate && e.startDate && s());
  });
  const Z = H(() => {
    var i;
    return (i = e.flow) != null && i.length && !e.partialFlow ? e.flowStep === e.flow.length : !0;
  }), le = () => {
    e.autoApply && Z.value && n("auto-apply", e.partialFlow);
  }, v = (i = !1) => {
    if (l.value)
      return Array.isArray(l.value) ? (o.value = l.value, d(i)) : P(l.value, i);
    if (p.value.count && i && !e.startDate)
      return D(S(), i);
  }, D = (i, V = !1) => {
    if ((!p.value.count || !p.value.static || V) && A(0, ge(i), fe(i)), p.value.count)
      for (let ne = 1; ne < p.value.count; ne++) {
        const me = De(S(), { month: U.value(ne - 1), year: W.value(ne - 1) }), we = ya(me, { months: 1 });
        c.value[ne] = { month: ge(we), year: fe(we) };
      }
  }, P = (i, V) => {
    D(i), B("hours", qe(i)), B("minutes", xe(i)), B("seconds", pt(i)), p.value.count && V && h();
  }, K = (i) => p.value.count ? Math.abs(qa(i[0], i[1])) >= p.value.count ? 1 : 0 : 1, f = (i, V) => {
    i[1] && e.showLastInRange ? D(i[K(i)], V) : D(i[0], V);
    const ne = (me, we) => [
      me(i[0]),
      i[1] ? me(i[1]) : b[we][1]
    ];
    B("hours", ne(qe, "hours")), B("minutes", ne(xe, "minutes")), B("seconds", ne(pt, "seconds"));
  }, E = (i, V) => {
    if ((e.range || e.weekPicker) && !e.multiDates)
      return f(i, V);
    if (e.multiDates) {
      const ne = i[i.length - 1];
      return P(ne, V);
    }
  }, d = (i) => {
    const V = l.value;
    E(V, i), p.value.count && p.value.solo && h();
  }, w = (i, V) => {
    const ne = De(S(), { month: U.value(V), year: W.value(V) }), me = i < 0 ? ct(ne, 1) : Pt(ne, 1);
    k(ge(me), fe(me), i < 0, e.preventMinMaxNavigation) && (A(V, ge(me), fe(me)), p.value.count && !p.value.solo && u(V), a());
  }, u = (i) => {
    for (let V = i - 1; V >= 0; V--) {
      const ne = Pt(De(S(), { month: U.value(V + 1), year: W.value(V + 1) }), 1);
      A(V, ge(ne), fe(ne));
    }
    for (let V = i + 1; V <= p.value.count - 1; V++) {
      const ne = ct(De(S(), { month: U.value(V - 1), year: W.value(V - 1) }), 1);
      A(V, ge(ne), fe(ne));
    }
  }, h = () => {
    if (Array.isArray(l.value) && l.value.length === 2) {
      const i = S(
        S(l.value[1] ? l.value[1] : ct(l.value[0], 1))
      ), [V, ne] = [ge(l.value[0]), fe(l.value[0])], [me, we] = [ge(l.value[1]), fe(l.value[1])];
      (V !== me || V === me && ne !== we) && p.value.solo && A(1, ge(i), fe(i));
    } else
      l.value && !Array.isArray(l.value) && (A(0, ge(l.value), fe(l.value)), D(S()));
  }, s = () => {
    e.startDate && (A(0, ge(S(e.startDate)), fe(S(e.startDate))), p.value.count && u(0));
  }, _ = (i, V) => {
    e.monthChangeOnScroll && w(e.monthChangeOnScroll !== "inverse" ? -i.deltaY : i.deltaY, V);
  }, te = (i, V, ne = !1) => {
    e.monthChangeOnArrows && e.vertical === ne && r(i, V);
  }, r = (i, V) => {
    w(i === "right" ? -1 : 1, V);
  }, L = (i) => e.markers.find((V) => ye(aa(i.value), aa(V.date))), C = (i, V) => {
    switch (e.sixWeeks === !0 ? "append" : e.sixWeeks) {
      case "prepend":
        return [!0, !1];
      case "center":
        return [i == 0, !0];
      case "fair":
        return [i == 0 || V > i, !0];
      case "append":
        return [!1, !1];
      default:
        return [!1, !1];
    }
  }, g = (i, V, ne, me) => {
    if (e.sixWeeks && i.length < 6) {
      const we = 6 - i.length, Qe = (V.getDay() + 7 - me) % 7, Ft = 6 - (ne.getDay() + 7 - me) % 7, [$t, rn] = C(Qe, Ft);
      for (let lt = 1; lt <= we; lt++)
        if (rn ? !!(lt % 2) == $t : $t) {
          const Vt = i[0].days[0], ln = F(ut(Vt.value, -7), ge(V));
          i.unshift({ days: ln });
        } else {
          const Vt = i[i.length - 1], ln = Vt.days[Vt.days.length - 1], Ia = F(ut(ln.value, 1), ge(V));
          i.push({ days: Ia });
        }
    }
    return i;
  }, F = (i, V) => {
    const ne = S(JSON.parse(JSON.stringify(i))), me = [];
    for (let we = 0; we < 7; we++) {
      const Qe = ut(ne, we), Mt = ge(Qe) !== V;
      me.push({
        text: e.hideOffsetDates && Mt ? "" : Qe.getDate(),
        value: Qe,
        current: !Mt,
        classData: {}
      });
    }
    return me;
  }, re = (i, V) => {
    const ne = [], me = S(Ze(new Date(V, i), e.timezone)), we = S(Ze(new Date(V, i + 1, 0), e.timezone)), Qe = e.weekStart, Mt = va(me, { weekStartsOn: Qe }), Ft = ($t) => {
      const rn = F($t, i);
      if (ne.push({ days: rn }), !ne[ne.length - 1].days.some(
        (lt) => ye(Le(lt.value), Le(we))
      )) {
        const lt = ut($t, 7);
        Ft(lt);
      }
    };
    return Ft(Mt), g(ne, me, we, Qe);
  }, G = (i) => (l.value = jt(S(i.value), e.timezone, e.weekStart), le()), ke = (i) => {
    const V = tt(S(i.value), b.hours, b.minutes, Et());
    e.multiDates ? Ln(V, l, e.multiDatesLimit) : l.value = V, t(), at().then(() => {
      le();
    });
  }, se = (i) => e.noDisabledRange ? Ta(o.value[0], i).some((ne) => T(ne)) : !1, R = () => {
    o.value = l.value ? l.value.slice() : [], o.value.length === 2 && !(e.fixedStart || e.fixedEnd) && (o.value = []);
  }, J = (i, V) => {
    const ne = [S(i.value), ut(S(i.value), +e.autoRange)];
    Y(ne) && (V && $e(i.value), o.value = ne);
  }, $e = (i) => {
    const V = ge(S(i)), ne = fe(S(i));
    if (A(0, V, ne), p.value.count > 0)
      for (let me = 1; me < p.value.count; me++) {
        const we = Rr(
          De(S(i), { year: U.value(me - 1), month: W.value(me - 1) })
        );
        A(me, we.month, we.year);
      }
  }, X = (i) => Array.isArray(l.value) && l.value.length === 2 ? e.fixedStart && (Ne(i, l.value[0]) || ye(i, l.value[0])) ? [l.value[0], i] : e.fixedEnd && (Pe(i, l.value[1]) || ye(i, l.value[1])) ? [i, l.value[1]] : (n("invalid-fixed-range", i), l.value) : [], Ve = (i) => {
    se(i.value) || !q(i.value, l.value, e.fixedStart ? 0 : 1) || (o.value = X(S(i.value)));
  }, _e = (i, V) => {
    if (R(), e.autoRange)
      return J(i, V);
    if (e.fixedStart || e.fixedEnd)
      return Ve(i);
    o.value[0] ? q(S(i.value), l.value) && !se(i.value) && (Pe(S(i.value), S(o.value[0])) ? (o.value.unshift(S(i.value)), n("range-end", o.value[0])) : (o.value[1] = S(i.value), n("range-end", o.value[1]))) : (o.value[0] = S(i.value), n("range-start", o.value[0]));
  }, Et = (i = !0) => e.enableSeconds ? Array.isArray(b.seconds) ? i ? b.seconds[0] : b.seconds[1] : b.seconds : 0, Dt = (i) => {
    o.value[i] = tt(
      o.value[i],
      b.hours[i],
      b.minutes[i],
      Et(i !== 1)
    );
  }, Jt = () => {
    var i, V;
    o.value[0] && o.value[1] && +((i = o.value) == null ? void 0 : i[0]) > +((V = o.value) == null ? void 0 : V[1]) && (o.value.reverse(), n("range-start", o.value[0]), n("range-end", o.value[1]));
  }, Xt = () => {
    o.value.length && (o.value[0] && !o.value[1] ? Dt(0) : (Dt(0), Dt(1), t()), Jt(), l.value = o.value.slice(), Un(o.value, n, e.autoApply, e.modelAuto));
  }, Qt = (i, V = !1) => {
    if (!(T(i.value) || !i.current && e.hideOffsetDates)) {
      if (e.weekPicker)
        return G(i);
      if (!e.range)
        return ke(i);
      sa(b.hours) && sa(b.minutes) && !e.multiDates && (_e(i, V), Xt());
    }
  }, en = (i, V) => {
    A(i, V.month, V.year), p.value.count && !p.value.solo && u(i), a(p.value.solo ? i : void 0), t();
  }, tn = (i, V) => {
    Array.isArray(i) && i.length <= 2 && e.range ? l.value = i.map((ne) => Ze(S(ne), V ? void 0 : e.timezone)) : Array.isArray(i) || (l.value = Ze(S(i), V ? void 0 : e.timezone)), x(), e.multiCalendars && at().then(() => v(!0));
  }, nn = () => {
    e.range ? l.value && Array.isArray(l.value) && l.value[0] ? l.value = Pe(S(), l.value[0]) ? [S(), l.value[0]] : [l.value[0], S()] : l.value = [S()] : l.value = S(), x();
  }, an = () => {
    if (Array.isArray(l.value))
      if (e.multiDates) {
        const i = ae();
        l.value[l.value.length - 1] = Q(i);
      } else
        l.value = l.value.map((i, V) => i && Q(i, V));
    else
      l.value = Q(l.value);
    n("time-update");
  }, ae = () => Array.isArray(l.value) && l.value.length ? l.value[l.value.length - 1] : null;
  return {
    calendars: c,
    modelValue: l,
    month: U,
    year: W,
    time: b,
    disabledTimesConfig: N,
    getCalendarDays: re,
    getMarker: L,
    handleScroll: _,
    handleSwipe: r,
    handleArrow: te,
    selectDate: Qt,
    updateMonthYear: en,
    presetDate: tn,
    selectCurrentDate: nn,
    updateTime: (i, V = !0, ne = !1) => {
      j(i, V, ne, an);
    }
  };
}, jl = { key: 0 }, Kl = /* @__PURE__ */ Be({
  __name: "DatePicker",
  props: {
    ...Xe
  },
  emits: [
    "tooltip-open",
    "tooltip-close",
    "mount",
    "update:internal-model-value",
    "update-flow-step",
    "reset-flow",
    "auto-apply",
    "focus-menu",
    "select-date",
    "range-start",
    "range-end",
    "invalid-fixed-range",
    "time-update",
    "am-pm-change",
    "time-picker-open",
    "time-picker-close",
    "recalculate-position"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, {
      calendars: o,
      month: l,
      year: c,
      modelValue: b,
      time: p,
      disabledTimesConfig: M,
      getCalendarDays: k,
      getMarker: T,
      handleArrow: Y,
      handleScroll: q,
      handleSwipe: j,
      selectDate: Q,
      updateMonthYear: B,
      presetDate: y,
      selectCurrentDate: N,
      updateTime: U
    } = zl(t, a, u, h), W = Ot(), { setHoverDate: A, getDayClassData: x, clearHoverDate: Z } = qr(b, t), { defaultedMultiCalendars: le } = Ce(t), v = z([]), D = z([]), P = z(null), K = je(W, "calendar"), f = je(W, "monthYear"), E = je(W, "timePicker"), d = (C) => {
      t.shadow || a("mount", C);
    };
    mt(
      o,
      () => {
        t.shadow || setTimeout(() => {
          a("recalculate-position");
        }, 0);
      },
      { deep: !0 }
    );
    const w = H(() => (C) => k(l.value(C), c.value(C)).map((g) => ({
      ...g,
      days: g.days.map((F) => (F.marker = T(F), F.classData = x(F), F))
    })));
    function u(C) {
      var g;
      C || C === 0 ? (g = D.value[C]) == null || g.triggerTransition(l.value(C), c.value(C)) : D.value.forEach((F, re) => F.triggerTransition(l.value(re), c.value(re)));
    }
    function h() {
      a("update-flow-step");
    }
    const s = (C, g = !1) => {
      Q(C, g), t.spaceConfirm && a("select-date");
    };
    return n({
      clearHoverDate: Z,
      presetDate: y,
      selectCurrentDate: N,
      toggleMonthPicker: (C, g, F = 0) => {
        var re;
        (re = v.value[F]) == null || re.toggleMonthPicker(C, g);
      },
      toggleYearPicker: (C, g, F = 0) => {
        var re;
        (re = v.value[F]) == null || re.toggleYearPicker(C, g);
      },
      toggleTimePicker: (C, g, F) => {
        var re;
        (re = P.value) == null || re.toggleTimePicker(C, g, F);
      },
      handleArrow: Y,
      updateMonthYear: B,
      getSidebarProps: () => ({
        modelValue: b,
        month: l,
        year: c,
        time: p,
        updateTime: U,
        updateMonthYear: B,
        selectDate: Q,
        presetDate: y
      })
    }), (C, g) => ($(), O(ce, null, [
      Ke(Hn, {
        "multi-calendars": m(le).count
      }, {
        default: de(({ instance: F, index: re }) => [
          C.disableMonthYearSelect ? I("", !0) : ($(), ve(Il, Ee({
            key: 0,
            ref: (G) => {
              G && (v.value[re] = G);
            },
            months: m(wa)(C.formatLocale, C.locale, C.monthNameFormat),
            years: m(Fn)(C.yearRange, C.reverseYears),
            month: m(l)(F),
            year: m(c)(F),
            instance: F
          }, C.$props, {
            onMount: g[0] || (g[0] = (G) => d(m(vt).header)),
            onResetFlow: g[1] || (g[1] = (G) => C.$emit("reset-flow")),
            onUpdateMonthYear: (G) => m(B)(F, G),
            onOverlayClosed: g[2] || (g[2] = (G) => C.$emit("focus-menu"))
          }), Fe({ _: 2 }, [
            Me(m(f), (G, ke) => ({
              name: G,
              fn: de((se) => [
                ee(C.$slots, G, Re(He(se)))
              ])
            }))
          ]), 1040, ["months", "years", "month", "year", "instance", "onUpdateMonthYear"])),
          Ke(Wl, Ee({
            ref: (G) => {
              G && (D.value[re] = G);
            },
            "mapped-dates": w.value(F),
            month: m(l)(F),
            year: m(c)(F)
          }, C.$props, {
            onSelectDate: (G) => m(Q)(G, F !== 1),
            onHandleSpace: (G) => s(G, F !== 1),
            onSetHoverDate: g[3] || (g[3] = (G) => m(A)(G)),
            onHandleScroll: (G) => m(q)(G, F),
            onHandleSwipe: (G) => m(j)(G, F),
            onMount: g[4] || (g[4] = (G) => d(m(vt).calendar)),
            onResetFlow: g[5] || (g[5] = (G) => C.$emit("reset-flow")),
            onTooltipOpen: g[6] || (g[6] = (G) => C.$emit("tooltip-open", G)),
            onTooltipClose: g[7] || (g[7] = (G) => C.$emit("tooltip-close", G))
          }), Fe({ _: 2 }, [
            Me(m(K), (G, ke) => ({
              name: G,
              fn: de((se) => [
                ee(C.$slots, G, Re(He({ ...se })))
              ])
            }))
          ]), 1040, ["mapped-dates", "month", "year", "onSelectDate", "onHandleSpace", "onHandleScroll", "onHandleSwipe"])
        ]),
        _: 3
      }, 8, ["multi-calendars"]),
      C.enableTimePicker ? ($(), O("div", jl, [
        C.$slots["time-picker"] ? ee(C.$slots, "time-picker", Re(Ee({ key: 0 }, { time: m(p), updateTime: m(U) }))) : ($(), ve(Ra, Ee({
          key: 1,
          ref_key: "timePickerRef",
          ref: P
        }, C.$props, {
          hours: m(p).hours,
          minutes: m(p).minutes,
          seconds: m(p).seconds,
          "internal-model-value": C.internalModelValue,
          "disabled-times-config": m(M),
          onMount: g[8] || (g[8] = (F) => d(m(vt).timePicker)),
          "onUpdate:hours": g[9] || (g[9] = (F) => m(U)(F)),
          "onUpdate:minutes": g[10] || (g[10] = (F) => m(U)(F, !1)),
          "onUpdate:seconds": g[11] || (g[11] = (F) => m(U)(F, !1, !0)),
          onResetFlow: g[12] || (g[12] = (F) => C.$emit("reset-flow")),
          onOverlayClosed: g[13] || (g[13] = (F) => C.$emit("time-picker-close")),
          onOverlayOpened: g[14] || (g[14] = (F) => C.$emit("time-picker-open", F)),
          onAmPmChange: g[15] || (g[15] = (F) => C.$emit("am-pm-change", F))
        }), Fe({ _: 2 }, [
          Me(m(E), (F, re) => ({
            name: F,
            fn: de((G) => [
              ee(C.$slots, F, Re(He(G)))
            ])
          }))
        ]), 1040, ["hours", "minutes", "seconds", "internal-model-value", "disabled-times-config"]))
      ])) : I("", !0)
    ], 64));
  }
}), Gl = ["id", "onKeydown"], Zl = {
  key: 0,
  class: "dp__sidebar_left"
}, ql = {
  key: 1,
  class: "dp--preset-dates"
}, xl = ["onClick", "onKeydown"], Jl = {
  key: 2,
  class: "dp__sidebar_right"
}, Xl = {
  key: 3,
  class: "dp__action_extra"
}, ua = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "DatepickerMenu",
  props: {
    ...xt,
    shadow: { type: Boolean, default: !1 },
    openOnTop: { type: Boolean, default: !1 },
    internalModelValue: { type: [Date, Array], default: null },
    arrMapValues: { type: Object, default: () => ({}) }
  },
  emits: [
    "close-picker",
    "select-date",
    "auto-apply",
    "time-update",
    "flow-step",
    "update-month-year",
    "invalid-select",
    "update:internal-model-value",
    "recalculate-position",
    "invalid-fixed-range",
    "tooltip-open",
    "tooltip-close",
    "time-picker-open",
    "time-picker-close",
    "am-pm-change",
    "range-start",
    "range-end"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, o = H(() => {
      const { openOnTop: R, ...J } = t;
      return {
        ...J,
        flowStep: x.value
      };
    }), { setMenuFocused: l, setShiftKey: c, control: b } = Pa(), p = Ot(), { defaultedTextInput: M, defaultedInline: k } = Ce(t), T = z(null), Y = z(0), q = z(null), j = z(null), Q = z(!1), B = z(null);
    Ue(() => {
      if (!t.shadow) {
        Q.value = !0, y(), window.addEventListener("resize", y);
        const R = Ae(q);
        if (R && !M.value.enabled && !k.value.enabled && (l(!0), P()), R) {
          const J = ($e) => {
            t.allowPreventDefault && $e.preventDefault(), $e.stopImmediatePropagation(), $e.stopPropagation();
          };
          R.addEventListener("pointerdown", J), R.addEventListener("mousedown", J);
        }
      }
    }), _n(() => {
      window.removeEventListener("resize", y);
    });
    const y = () => {
      const R = Ae(j);
      R && (Y.value = R.getBoundingClientRect().width);
    }, { arrowRight: N, arrowLeft: U, arrowDown: W, arrowUp: A } = rt(), { flowStep: x, updateFlowStep: Z, childMount: le, resetFlow: v } = xr(t, a, B), D = H(() => t.monthPicker ? vl : t.yearPicker ? gl : t.timePicker ? Rl : Kl), P = () => {
      const R = Ae(q);
      R && R.focus({ preventScroll: !0 });
    }, K = () => H(() => {
      var R;
      return (R = B.value) == null ? void 0 : R.getSidebarProps();
    }), f = () => {
      t.openOnTop && a("recalculate-position");
    }, E = je(p, "action"), d = H(() => t.monthPicker || t.yearPicker ? je(p, "monthYear") : t.timePicker ? je(p, "timePicker") : je(p, "shared")), w = H(() => t.openOnTop ? "dp__arrow_bottom" : "dp__arrow_top"), u = H(() => ({
      dp__menu_disabled: t.disabled,
      dp__menu_readonly: t.readonly
    })), h = H(
      () => ({
        dp__menu: !0,
        dp__menu_index: !k.value.enabled,
        dp__relative: k.value.enabled,
        [t.menuClassName]: !!t.menuClassName
      })
    ), s = (R) => {
      R.stopPropagation(), R.stopImmediatePropagation();
    }, _ = () => {
      t.escClose && a("close-picker");
    }, te = (R) => {
      if (t.arrowNavigation) {
        if (R === "up")
          return A();
        if (R === "down")
          return W();
        if (R === "left")
          return U();
        if (R === "right")
          return N();
      } else
        R === "left" || R === "up" ? F("handleArrow", "left", 0, R === "up") : F("handleArrow", "right", 0, R === "down");
    }, r = (R) => {
      c(R.shiftKey), !t.disableMonthYearSelect && R.code === "Tab" && R.target.classList.contains("dp__menu") && b.value.shiftKeyInMenu && (R.preventDefault(), R.stopImmediatePropagation(), a("close-picker"));
    }, L = () => {
      P(), a("time-picker-close");
    }, C = (R) => {
      var J, $e, X;
      (J = B.value) == null || J.toggleTimePicker(!1, !1), ($e = B.value) == null || $e.toggleMonthPicker(!1, !1, R), (X = B.value) == null || X.toggleYearPicker(!1, !1, R);
    }, g = (R, J = 0) => {
      var $e, X, Ve;
      return R === "month" ? ($e = B.value) == null ? void 0 : $e.toggleMonthPicker(!1, !0, J) : R === "year" ? (X = B.value) == null ? void 0 : X.toggleYearPicker(!1, !0, J) : R === "time" ? (Ve = B.value) == null ? void 0 : Ve.toggleTimePicker(!0, !1) : C(J);
    }, F = (R, ...J) => {
      var $e, X;
      ($e = B.value) != null && $e[R] && ((X = B.value) == null || X[R](...J));
    }, re = () => {
      F("selectCurrentDate");
    }, G = (R, J) => {
      F("presetDate", R, J);
    }, ke = () => {
      F("clearHoverDate");
    };
    return n({
      updateMonthYear: (R, J) => {
        F("updateMonthYear", R, J);
      },
      switchView: g
    }), (R, J) => {
      var $e;
      return $(), O("div", {
        id: R.uid ? `dp-menu-${R.uid}` : void 0,
        tabindex: "0",
        ref_key: "dpMenuRef",
        ref: q,
        role: "dialog",
        class: he(h.value),
        onMouseleave: ke,
        onClick: s,
        onKeydown: [
          oe(_, ["esc"]),
          J[14] || (J[14] = oe(pe((X) => te("left"), ["prevent"]), ["left"])),
          J[15] || (J[15] = oe(pe((X) => te("up"), ["prevent"]), ["up"])),
          J[16] || (J[16] = oe(pe((X) => te("down"), ["prevent"]), ["down"])),
          J[17] || (J[17] = oe(pe((X) => te("right"), ["prevent"]), ["right"])),
          r
        ]
      }, [
        (R.disabled || R.readonly) && m(k).enabled ? ($(), O("div", {
          key: 0,
          class: he(u.value)
        }, null, 2)) : I("", !0),
        !m(k).enabled && !R.teleportCenter ? ($(), O("div", {
          key: 1,
          class: he(w.value)
        }, null, 2)) : I("", !0),
        ue("div", {
          ref_key: "innerMenuRef",
          ref: j,
          class: he({
            dp__menu_content_wrapper: (($e = R.presetDates) == null ? void 0 : $e.length) || !!R.$slots["left-sidebar"] || !!R.$slots["right-sidebar"]
          }),
          style: Ge({ "--dp-menu-width": `${Y.value}px` })
        }, [
          R.$slots["left-sidebar"] ? ($(), O("div", Zl, [
            ee(R.$slots, "left-sidebar", Re(He(K())))
          ])) : I("", !0),
          R.presetDates.length ? ($(), O("div", ql, [
            ($(!0), O(ce, null, Me(R.presetDates, (X, Ve) => ($(), O("div", {
              key: Ve,
              style: Ge(X.style || {}),
              class: "dp--preset-range"
            }, [
              X.slot ? ee(R.$slots, X.slot, {
                key: 0,
                presetDate: G,
                label: X.label,
                value: X.value
              }) : ($(), O("div", {
                key: 1,
                role: "button",
                tabindex: "0",
                onClick: (_e) => G(X.value, X.noTz),
                onKeydown: [
                  oe(pe((_e) => G(X.value, X.noTz), ["prevent"]), ["enter"]),
                  oe(pe((_e) => G(X.value, X.noTz), ["prevent"]), ["space"])
                ]
              }, Ie(X.label), 41, xl))
            ], 4))), 128))
          ])) : I("", !0),
          ue("div", {
            class: "dp__instance_calendar",
            ref_key: "calendarWrapperRef",
            ref: T,
            role: "document"
          }, [
            ($(), ve(Sn(D.value), Ee({
              ref_key: "dynCmpRef",
              ref: B
            }, o.value, {
              "flow-step": m(x),
              onMount: m(le),
              onUpdateFlowStep: m(Z),
              onResetFlow: m(v),
              onFocusMenu: P,
              onSelectDate: J[0] || (J[0] = (X) => R.$emit("select-date")),
              onTooltipOpen: J[1] || (J[1] = (X) => R.$emit("tooltip-open", X)),
              onTooltipClose: J[2] || (J[2] = (X) => R.$emit("tooltip-close", X)),
              onAutoApply: J[3] || (J[3] = (X) => R.$emit("auto-apply", X)),
              onRangeStart: J[4] || (J[4] = (X) => R.$emit("range-start", X)),
              onRangeEnd: J[5] || (J[5] = (X) => R.$emit("range-end", X)),
              onInvalidFixedRange: J[6] || (J[6] = (X) => R.$emit("invalid-fixed-range", X)),
              onTimeUpdate: J[7] || (J[7] = (X) => R.$emit("time-update")),
              onAmPmChange: J[8] || (J[8] = (X) => R.$emit("am-pm-change", X)),
              onTimePickerOpen: J[9] || (J[9] = (X) => R.$emit("time-picker-open", X)),
              onTimePickerClose: L,
              onRecalculatePosition: f,
              "onUpdate:internalModelValue": J[10] || (J[10] = (X) => R.$emit("update:internal-model-value", X))
            }), Fe({ _: 2 }, [
              Me(d.value, (X, Ve) => ({
                name: X,
                fn: de((_e) => [
                  ee(R.$slots, X, Re(He({ ..._e })))
                ])
              }))
            ]), 1040, ["flow-step", "onMount", "onUpdateFlowStep", "onResetFlow"]))
          ], 512),
          R.$slots["right-sidebar"] ? ($(), O("div", Jl, [
            ee(R.$slots, "right-sidebar", Re(He(K())))
          ])) : I("", !0),
          R.$slots["action-extra"] ? ($(), O("div", Xl, [
            R.$slots["action-extra"] ? ee(R.$slots, "action-extra", {
              key: 0,
              selectCurrentDate: re
            }) : I("", !0)
          ])) : I("", !0)
        ], 6),
        !R.autoApply || R.keepActionRow ? ($(), ve(rl, Ee({
          key: 2,
          "menu-mount": Q.value
        }, o.value, {
          "calendar-width": Y.value,
          onClosePicker: J[11] || (J[11] = (X) => R.$emit("close-picker")),
          onSelectDate: J[12] || (J[12] = (X) => R.$emit("select-date")),
          onInvalidSelect: J[13] || (J[13] = (X) => R.$emit("invalid-select")),
          onSelectNow: re
        }), Fe({ _: 2 }, [
          Me(m(E), (X, Ve) => ({
            name: X,
            fn: de((_e) => [
              ee(R.$slots, X, Re(He({ ..._e })))
            ])
          }))
        ]), 1040, ["menu-mount", "calendar-width"])) : I("", !0)
      ], 42, Gl);
    };
  }
}), Ql = typeof window < "u" ? window : void 0, pn = () => {
}, eo = (e) => Ea() ? (Fa(e), !0) : !1, to = (e, n, a, t) => {
  if (!e)
    return pn;
  let o = pn;
  const l = mt(
    () => m(e),
    (b) => {
      o(), b && (b.addEventListener(n, a, t), o = () => {
        b.removeEventListener(n, a, t), o = pn;
      });
    },
    { immediate: !0, flush: "post" }
  ), c = () => {
    l(), o();
  };
  return eo(c), c;
}, no = (e, n, a, t = {}) => {
  const { window: o = Ql, event: l = "pointerdown" } = t;
  return o ? to(o, l, (b) => {
    const p = Ae(e), M = Ae(n);
    !p || !M || p === b.target || b.composedPath().includes(p) || b.composedPath().includes(M) || a(b);
  }, { passive: !0 }) : void 0;
}, ao = /* @__PURE__ */ Be({
  compatConfig: {
    MODE: 3
  },
  __name: "VueDatePicker",
  props: {
    ...xt
  },
  emits: [
    "update:model-value",
    "update:model-timezone-value",
    "text-submit",
    "closed",
    "cleared",
    "open",
    "focus",
    "blur",
    "internal-model-change",
    "recalculate-position",
    "flow-step",
    "update-month-year",
    "invalid-select",
    "invalid-fixed-range",
    "tooltip-open",
    "tooltip-close",
    "time-picker-open",
    "time-picker-close",
    "am-pm-change",
    "range-start",
    "range-end"
  ],
  setup(e, { expose: n, emit: a }) {
    const t = e, o = Ot(), l = z(!1), c = Kt(t, "modelValue"), b = Kt(t, "timezone"), p = z(null), M = z(null), k = z(null), T = z(!1), Y = z(null), { setMenuFocused: q, setShiftKey: j } = Pa(), { clearArrowNav: Q } = rt(), { mapDatesArrToMap: B, validateDate: y, isValidTime: N } = Bt(t), { defaultedTransitions: U, defaultedTextInput: W, defaultedInline: A } = Ce(t), { menuTransition: x, showTransition: Z } = Yt(U);
    Ue(() => {
      s(t.modelValue), at().then(() => {
        A.value.enabled || (d(Y.value).addEventListener("scroll", F), window.addEventListener("resize", re));
      }), A.value.enabled && (l.value = !0);
    });
    const le = H(() => B());
    _n(() => {
      if (!A.value.enabled) {
        const ae = d(Y.value);
        ae && ae.removeEventListener("scroll", F), window.removeEventListener("resize", re);
      }
    });
    const v = je(o, "all", t.presetDates), D = je(o, "input");
    mt(
      [c, b],
      () => {
        s(c.value);
      },
      { deep: !0 }
    );
    const { openOnTop: P, menuStyle: K, xCorrect: f, setMenuPosition: E, getScrollableParent: d, shadowRender: w } = Kr(
      p,
      M,
      k,
      Y,
      A,
      a,
      t
    ), {
      inputValue: u,
      internalModelValue: h,
      parseExternalModelValue: s,
      emitModelValue: _,
      formatInputValue: te,
      checkBeforeEmit: r
    } = zr(a, t, T), L = H(
      () => ({
        dp__main: !0,
        dp__theme_dark: t.dark,
        dp__theme_light: !t.dark,
        dp__flex_display: A.value.enabled,
        dp__flex_display_with_input: A.value.input
      })
    ), C = H(() => t.dark ? "dp__theme_dark" : "dp__theme_light"), g = H(() => t.teleport ? {
      to: typeof t.teleport == "boolean" ? "body" : t.teleport,
      disabled: A.value.enabled
    } : { class: "dp__outer_menu_wrap" }), F = () => {
      l.value && (t.closeOnScroll ? _e() : E());
    }, re = () => {
      l.value && E();
    }, G = () => {
      !t.disabled && !t.readonly && (w(ua, t), E(!1), l.value = !0, l.value && a("open"), l.value || Ve(), s(t.modelValue));
    }, ke = () => {
      u.value = "", Ve(), a("update:model-value", null), a("update:model-timezone-value", null), a("cleared"), t.closeOnClearValue && _e();
    }, se = () => {
      const ae = h.value;
      return !ae || !Array.isArray(ae) && y(ae) ? !0 : Array.isArray(ae) ? ae.length === 2 && y(ae[0]) && y(ae[1]) ? !0 : t.partialRange && !t.timePicker ? y(ae[0]) : !1 : !1;
    }, R = () => {
      r() && se() ? (_(), _e()) : a("invalid-select", h.value);
    }, J = (ae) => {
      $e(), _(), t.closeOnAutoApply && !ae && _e();
    }, $e = () => {
      k.value && W.value.enabled && k.value.setParsedDate(h.value);
    }, X = (ae = !1) => {
      t.autoApply && N(h.value) && se() && (t.range && Array.isArray(h.value) ? (t.partialRange || h.value.length === 2) && J(ae) : J(ae));
    }, Ve = () => {
      W.value.enabled || (h.value = null);
    }, _e = () => {
      A.value.enabled || (l.value && (l.value = !1, f.value = !1, q(!1), j(!1), Q(), a("closed"), u.value && s(c.value)), Ve(), a("blur"));
    }, Et = (ae, ie) => {
      if (!ae) {
        h.value = null;
        return;
      }
      h.value = ae, ie && (R(), a("text-submit"));
    }, Dt = () => {
      t.autoApply && N(h.value) && _(), $e();
    }, Jt = () => l.value ? _e() : G(), Xt = (ae) => {
      h.value = ae;
    }, Qt = () => {
      W.value.enabled && (T.value = !0, te()), a("focus");
    }, en = () => {
      W.value.enabled && (T.value = !1, s(t.modelValue)), a("blur");
    }, tn = (ae) => {
      M.value && M.value.updateMonthYear(0, {
        month: ea(ae.month),
        year: ea(ae.year)
      });
    }, nn = (ae) => {
      s(ae ?? t.modelValue);
    }, an = (ae, ie) => {
      var i;
      (i = M.value) == null || i.switchView(ae, ie);
    };
    return no(
      p,
      k,
      t.onClickOutside ? () => t.onClickOutside(se) : _e
    ), n({
      closeMenu: _e,
      selectDate: R,
      clearValue: ke,
      openMenu: G,
      onScroll: F,
      formatInputValue: te,
      // exposed for testing purposes
      updateInternalModelValue: Xt,
      // modify internal modelValue
      setMonthYear: tn,
      parseModel: nn,
      switchView: an
    }), (ae, ie) => ($(), O("div", {
      class: he(L.value),
      ref_key: "pickerWrapperRef",
      ref: Y
    }, [
      Ke(el, Ee({
        ref_key: "inputRef",
        ref: k,
        "is-menu-open": l.value,
        "input-value": m(u),
        "onUpdate:inputValue": ie[0] || (ie[0] = (i) => zn(u) ? u.value = i : null)
      }, ae.$props, {
        onClear: ke,
        onOpen: G,
        onSetInputDate: Et,
        onSetEmptyDate: m(_),
        onSelectDate: R,
        onToggle: Jt,
        onClose: _e,
        onFocus: Qt,
        onBlur: en,
        onRealBlur: ie[1] || (ie[1] = (i) => T.value = !1)
      }), Fe({ _: 2 }, [
        Me(m(D), (i, V) => ({
          name: i,
          fn: de((ne) => [
            ee(ae.$slots, i, Re(He(ne)))
          ])
        }))
      ]), 1040, ["is-menu-open", "input-value", "onSetEmptyDate"]),
      Ke(wt, {
        name: m(x)(m(P)),
        css: m(Z) && !m(A).enabled
      }, {
        default: de(() => [
          l.value ? ($(), ve(Sn(ae.teleport ? Va : "div"), Ee({
            key: 0,
            ref_key: "dpWrapMenuRef",
            ref: p
          }, g.value, {
            class: { "dp--menu-wrapper": !m(A).enabled },
            style: m(A).enabled ? void 0 : m(K)
          }), {
            default: de(() => [
              Ke(ua, Ee({
                ref_key: "dpMenuRef",
                ref: M,
                class: { [C.value]: !0, "dp--menu-wrapper": ae.teleport },
                style: ae.teleport ? m(K) : void 0,
                "open-on-top": m(P),
                "arr-map-values": le.value
              }, ae.$props, {
                "internal-model-value": m(h),
                "onUpdate:internalModelValue": ie[2] || (ie[2] = (i) => zn(h) ? h.value = i : null),
                onClosePicker: _e,
                onSelectDate: R,
                onAutoApply: X,
                onTimeUpdate: Dt,
                onFlowStep: ie[3] || (ie[3] = (i) => ae.$emit("flow-step", i)),
                onUpdateMonthYear: ie[4] || (ie[4] = (i) => ae.$emit("update-month-year", i)),
                onInvalidSelect: ie[5] || (ie[5] = (i) => ae.$emit("invalid-select", m(h))),
                onInvalidFixedRange: ie[6] || (ie[6] = (i) => ae.$emit("invalid-fixed-range", i)),
                onRecalculatePosition: m(E),
                onTooltipOpen: ie[7] || (ie[7] = (i) => ae.$emit("tooltip-open", i)),
                onTooltipClose: ie[8] || (ie[8] = (i) => ae.$emit("tooltip-close", i)),
                onTimePickerOpen: ie[9] || (ie[9] = (i) => ae.$emit("time-picker-open", i)),
                onTimePickerClose: ie[10] || (ie[10] = (i) => ae.$emit("time-picker-close", i)),
                onAmPmChange: ie[11] || (ie[11] = (i) => ae.$emit("am-pm-change", i)),
                onRangeStart: ie[12] || (ie[12] = (i) => ae.$emit("range-start", i)),
                onRangeEnd: ie[13] || (ie[13] = (i) => ae.$emit("range-end", i))
              }), Fe({ _: 2 }, [
                Me(m(v), (i, V) => ({
                  name: i,
                  fn: de((ne) => [
                    ee(ae.$slots, i, Re(He({ ...ne })))
                  ])
                }))
              ]), 1040, ["class", "style", "open-on-top", "arr-map-values", "internal-model-value", "onRecalculatePosition"])
            ]),
            _: 3
          }, 16, ["class", "style"])) : I("", !0)
        ]),
        _: 3
      }, 8, ["name", "css"])
    ], 2));
  }
}), Oa = /* @__PURE__ */ (() => {
  const e = ao;
  return e.install = (n) => {
    n.component("Vue3DatePicker", e);
  }, e;
})(), ro = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Oa
}, Symbol.toStringTag, { value: "Module" }));
Object.entries(ro).forEach(([e, n]) => {
  e !== "default" && (Oa[e] = n);
});
export {
  Oa as default
};
